\documentclass[11pt]{article}

    \usepackage[breakable]{tcolorbox}
    \usepackage{parskip} % Stop auto-indenting (to mimic markdown behaviour)
    

    % Basic figure setup, for now with no caption control since it's done
    % automatically by Pandoc (which extracts ![](path) syntax from Markdown).
    \usepackage{graphicx}
    % Maintain compatibility with old templates. Remove in nbconvert 6.0
    \let\Oldincludegraphics\includegraphics
    % Ensure that by default, figures have no caption (until we provide a
    % proper Figure object with a Caption API and a way to capture that
    % in the conversion process - todo).
    \usepackage{caption}
    \DeclareCaptionFormat{nocaption}{}
    \captionsetup{format=nocaption,aboveskip=0pt,belowskip=0pt}

    \usepackage{float}
    \floatplacement{figure}{H} % forces figures to be placed at the correct location
    \usepackage{xcolor} % Allow colors to be defined
    \usepackage{enumerate} % Needed for markdown enumerations to work
    \usepackage{geometry} % Used to adjust the document margins
    \usepackage{amsmath} % Equations
    \usepackage{amssymb} % Equations
    \usepackage{textcomp} % defines textquotesingle
    % Hack from http://tex.stackexchange.com/a/47451/13684:
    \AtBeginDocument{%
        \def\PYZsq{\textquotesingle}% Upright quotes in Pygmentized code
    }
    \usepackage{upquote} % Upright quotes for verbatim code
    \usepackage{eurosym} % defines \euro

    \usepackage{iftex}
    \ifPDFTeX
        \usepackage[T1]{fontenc}
        \IfFileExists{alphabeta.sty}{
              \usepackage{alphabeta}
          }{
              \usepackage[mathletters]{ucs}
              \usepackage[utf8x]{inputenc}
          }
    \else
        \usepackage{fontspec}
        \usepackage{unicode-math}
    \fi

    \usepackage{fancyvrb} % verbatim replacement that allows latex
    \usepackage{grffile} % extends the file name processing of package graphics
                         % to support a larger range
    \makeatletter % fix for old versions of grffile with XeLaTeX
    \@ifpackagelater{grffile}{2019/11/01}
    {
      % Do nothing on new versions
    }
    {
      \def\Gread@@xetex#1{%
        \IfFileExists{"\Gin@base".bb}%
        {\Gread@eps{\Gin@base.bb}}%
        {\Gread@@xetex@aux#1}%
      }
    }
    \makeatother
    \usepackage[Export]{adjustbox} % Used to constrain images to a maximum size
    \adjustboxset{max size={0.9\linewidth}{0.9\paperheight}}

    % The hyperref package gives us a pdf with properly built
    % internal navigation ('pdf bookmarks' for the table of contents,
    % internal cross-reference links, web links for URLs, etc.)
    \usepackage{hyperref}
    % The default LaTeX title has an obnoxious amount of whitespace. By default,
    % titling removes some of it. It also provides customization options.
    \usepackage{titling}
    \usepackage{longtable} % longtable support required by pandoc >1.10
    \usepackage{booktabs}  % table support for pandoc > 1.12.2
    \usepackage{array}     % table support for pandoc >= 2.11.3
    \usepackage{calc}      % table minipage width calculation for pandoc >= 2.11.1
    \usepackage[inline]{enumitem} % IRkernel/repr support (it uses the enumerate* environment)
    \usepackage[normalem]{ulem} % ulem is needed to support strikethroughs (\sout)
                                % normalem makes italics be italics, not underlines
    \usepackage{mathrsfs}
    

    
    % Colors for the hyperref package
    \definecolor{urlcolor}{rgb}{0,.145,.698}
    \definecolor{linkcolor}{rgb}{.71,0.21,0.01}
    \definecolor{citecolor}{rgb}{.12,.54,.11}

    % ANSI colors
    \definecolor{ansi-black}{HTML}{3E424D}
    \definecolor{ansi-black-intense}{HTML}{282C36}
    \definecolor{ansi-red}{HTML}{E75C58}
    \definecolor{ansi-red-intense}{HTML}{B22B31}
    \definecolor{ansi-green}{HTML}{00A250}
    \definecolor{ansi-green-intense}{HTML}{007427}
    \definecolor{ansi-yellow}{HTML}{DDB62B}
    \definecolor{ansi-yellow-intense}{HTML}{B27D12}
    \definecolor{ansi-blue}{HTML}{208FFB}
    \definecolor{ansi-blue-intense}{HTML}{0065CA}
    \definecolor{ansi-magenta}{HTML}{D160C4}
    \definecolor{ansi-magenta-intense}{HTML}{A03196}
    \definecolor{ansi-cyan}{HTML}{60C6C8}
    \definecolor{ansi-cyan-intense}{HTML}{258F8F}
    \definecolor{ansi-white}{HTML}{C5C1B4}
    \definecolor{ansi-white-intense}{HTML}{A1A6B2}
    \definecolor{ansi-default-inverse-fg}{HTML}{FFFFFF}
    \definecolor{ansi-default-inverse-bg}{HTML}{000000}

    % common color for the border for error outputs.
    \definecolor{outerrorbackground}{HTML}{FFDFDF}

    % commands and environments needed by pandoc snippets
    % extracted from the output of `pandoc -s`
    \providecommand{\tightlist}{%
      \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
    \DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
    % Add ',fontsize=\small' for more characters per line
    \newenvironment{Shaded}{}{}
    \newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.00,0.44,0.13}{\textbf{{#1}}}}
    \newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.56,0.13,0.00}{{#1}}}
    \newcommand{\DecValTok}[1]{\textcolor[rgb]{0.25,0.63,0.44}{{#1}}}
    \newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.25,0.63,0.44}{{#1}}}
    \newcommand{\FloatTok}[1]{\textcolor[rgb]{0.25,0.63,0.44}{{#1}}}
    \newcommand{\CharTok}[1]{\textcolor[rgb]{0.25,0.44,0.63}{{#1}}}
    \newcommand{\StringTok}[1]{\textcolor[rgb]{0.25,0.44,0.63}{{#1}}}
    \newcommand{\CommentTok}[1]{\textcolor[rgb]{0.38,0.63,0.69}{\textit{{#1}}}}
    \newcommand{\OtherTok}[1]{\textcolor[rgb]{0.00,0.44,0.13}{{#1}}}
    \newcommand{\AlertTok}[1]{\textcolor[rgb]{1.00,0.00,0.00}{\textbf{{#1}}}}
    \newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.02,0.16,0.49}{{#1}}}
    \newcommand{\RegionMarkerTok}[1]{{#1}}
    \newcommand{\ErrorTok}[1]{\textcolor[rgb]{1.00,0.00,0.00}{\textbf{{#1}}}}
    \newcommand{\NormalTok}[1]{{#1}}

    % Additional commands for more recent versions of Pandoc
    \newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.53,0.00,0.00}{{#1}}}
    \newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.25,0.44,0.63}{{#1}}}
    \newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.25,0.44,0.63}{{#1}}}
    \newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.73,0.40,0.53}{{#1}}}
    \newcommand{\ImportTok}[1]{{#1}}
    \newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.73,0.13,0.13}{\textit{{#1}}}}
    \newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.38,0.63,0.69}{\textbf{\textit{{#1}}}}}
    \newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.38,0.63,0.69}{\textbf{\textit{{#1}}}}}
    \newcommand{\VariableTok}[1]{\textcolor[rgb]{0.10,0.09,0.49}{{#1}}}
    \newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.00,0.44,0.13}{\textbf{{#1}}}}
    \newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.40,0.40,0.40}{{#1}}}
    \newcommand{\BuiltInTok}[1]{{#1}}
    \newcommand{\ExtensionTok}[1]{{#1}}
    \newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.74,0.48,0.00}{{#1}}}
    \newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.49,0.56,0.16}{{#1}}}
    \newcommand{\InformationTok}[1]{\textcolor[rgb]{0.38,0.63,0.69}{\textbf{\textit{{#1}}}}}
    \newcommand{\WarningTok}[1]{\textcolor[rgb]{0.38,0.63,0.69}{\textbf{\textit{{#1}}}}}


    % Define a nice break command that doesn't care if a line doesn't already
    % exist.
    \def\br{\hspace*{\fill} \\* }
    % Math Jax compatibility definitions
    \def\gt{>}
    \def\lt{<}
    \let\Oldtex\TeX
    \let\Oldlatex\LaTeX
    \renewcommand{\TeX}{\textrm{\Oldtex}}
    \renewcommand{\LaTeX}{\textrm{\Oldlatex}}
    % Document parameters
    % Document title
    \title{ass2}
    
    
    
    
    
% Pygments definitions
\makeatletter
\def\PY@reset{\let\PY@it=\relax \let\PY@bf=\relax%
    \let\PY@ul=\relax \let\PY@tc=\relax%
    \let\PY@bc=\relax \let\PY@ff=\relax}
\def\PY@tok#1{\csname PY@tok@#1\endcsname}
\def\PY@toks#1+{\ifx\relax#1\empty\else%
    \PY@tok{#1}\expandafter\PY@toks\fi}
\def\PY@do#1{\PY@bc{\PY@tc{\PY@ul{%
    \PY@it{\PY@bf{\PY@ff{#1}}}}}}}
\def\PY#1#2{\PY@reset\PY@toks#1+\relax+\PY@do{#2}}

\@namedef{PY@tok@w}{\def\PY@tc##1{\textcolor[rgb]{0.73,0.73,0.73}{##1}}}
\@namedef{PY@tok@c}{\let\PY@it=\textit\def\PY@tc##1{\textcolor[rgb]{0.24,0.48,0.48}{##1}}}
\@namedef{PY@tok@cp}{\def\PY@tc##1{\textcolor[rgb]{0.61,0.40,0.00}{##1}}}
\@namedef{PY@tok@k}{\let\PY@bf=\textbf\def\PY@tc##1{\textcolor[rgb]{0.00,0.50,0.00}{##1}}}
\@namedef{PY@tok@kp}{\def\PY@tc##1{\textcolor[rgb]{0.00,0.50,0.00}{##1}}}
\@namedef{PY@tok@kt}{\def\PY@tc##1{\textcolor[rgb]{0.69,0.00,0.25}{##1}}}
\@namedef{PY@tok@o}{\def\PY@tc##1{\textcolor[rgb]{0.40,0.40,0.40}{##1}}}
\@namedef{PY@tok@ow}{\let\PY@bf=\textbf\def\PY@tc##1{\textcolor[rgb]{0.67,0.13,1.00}{##1}}}
\@namedef{PY@tok@nb}{\def\PY@tc##1{\textcolor[rgb]{0.00,0.50,0.00}{##1}}}
\@namedef{PY@tok@nf}{\def\PY@tc##1{\textcolor[rgb]{0.00,0.00,1.00}{##1}}}
\@namedef{PY@tok@nc}{\let\PY@bf=\textbf\def\PY@tc##1{\textcolor[rgb]{0.00,0.00,1.00}{##1}}}
\@namedef{PY@tok@nn}{\let\PY@bf=\textbf\def\PY@tc##1{\textcolor[rgb]{0.00,0.00,1.00}{##1}}}
\@namedef{PY@tok@ne}{\let\PY@bf=\textbf\def\PY@tc##1{\textcolor[rgb]{0.80,0.25,0.22}{##1}}}
\@namedef{PY@tok@nv}{\def\PY@tc##1{\textcolor[rgb]{0.10,0.09,0.49}{##1}}}
\@namedef{PY@tok@no}{\def\PY@tc##1{\textcolor[rgb]{0.53,0.00,0.00}{##1}}}
\@namedef{PY@tok@nl}{\def\PY@tc##1{\textcolor[rgb]{0.46,0.46,0.00}{##1}}}
\@namedef{PY@tok@ni}{\let\PY@bf=\textbf\def\PY@tc##1{\textcolor[rgb]{0.44,0.44,0.44}{##1}}}
\@namedef{PY@tok@na}{\def\PY@tc##1{\textcolor[rgb]{0.41,0.47,0.13}{##1}}}
\@namedef{PY@tok@nt}{\let\PY@bf=\textbf\def\PY@tc##1{\textcolor[rgb]{0.00,0.50,0.00}{##1}}}
\@namedef{PY@tok@nd}{\def\PY@tc##1{\textcolor[rgb]{0.67,0.13,1.00}{##1}}}
\@namedef{PY@tok@s}{\def\PY@tc##1{\textcolor[rgb]{0.73,0.13,0.13}{##1}}}
\@namedef{PY@tok@sd}{\let\PY@it=\textit\def\PY@tc##1{\textcolor[rgb]{0.73,0.13,0.13}{##1}}}
\@namedef{PY@tok@si}{\let\PY@bf=\textbf\def\PY@tc##1{\textcolor[rgb]{0.64,0.35,0.47}{##1}}}
\@namedef{PY@tok@se}{\let\PY@bf=\textbf\def\PY@tc##1{\textcolor[rgb]{0.67,0.36,0.12}{##1}}}
\@namedef{PY@tok@sr}{\def\PY@tc##1{\textcolor[rgb]{0.64,0.35,0.47}{##1}}}
\@namedef{PY@tok@ss}{\def\PY@tc##1{\textcolor[rgb]{0.10,0.09,0.49}{##1}}}
\@namedef{PY@tok@sx}{\def\PY@tc##1{\textcolor[rgb]{0.00,0.50,0.00}{##1}}}
\@namedef{PY@tok@m}{\def\PY@tc##1{\textcolor[rgb]{0.40,0.40,0.40}{##1}}}
\@namedef{PY@tok@gh}{\let\PY@bf=\textbf\def\PY@tc##1{\textcolor[rgb]{0.00,0.00,0.50}{##1}}}
\@namedef{PY@tok@gu}{\let\PY@bf=\textbf\def\PY@tc##1{\textcolor[rgb]{0.50,0.00,0.50}{##1}}}
\@namedef{PY@tok@gd}{\def\PY@tc##1{\textcolor[rgb]{0.63,0.00,0.00}{##1}}}
\@namedef{PY@tok@gi}{\def\PY@tc##1{\textcolor[rgb]{0.00,0.52,0.00}{##1}}}
\@namedef{PY@tok@gr}{\def\PY@tc##1{\textcolor[rgb]{0.89,0.00,0.00}{##1}}}
\@namedef{PY@tok@ge}{\let\PY@it=\textit}
\@namedef{PY@tok@gs}{\let\PY@bf=\textbf}
\@namedef{PY@tok@gp}{\let\PY@bf=\textbf\def\PY@tc##1{\textcolor[rgb]{0.00,0.00,0.50}{##1}}}
\@namedef{PY@tok@go}{\def\PY@tc##1{\textcolor[rgb]{0.44,0.44,0.44}{##1}}}
\@namedef{PY@tok@gt}{\def\PY@tc##1{\textcolor[rgb]{0.00,0.27,0.87}{##1}}}
\@namedef{PY@tok@err}{\def\PY@bc##1{{\setlength{\fboxsep}{\string -\fboxrule}\fcolorbox[rgb]{1.00,0.00,0.00}{1,1,1}{\strut ##1}}}}
\@namedef{PY@tok@kc}{\let\PY@bf=\textbf\def\PY@tc##1{\textcolor[rgb]{0.00,0.50,0.00}{##1}}}
\@namedef{PY@tok@kd}{\let\PY@bf=\textbf\def\PY@tc##1{\textcolor[rgb]{0.00,0.50,0.00}{##1}}}
\@namedef{PY@tok@kn}{\let\PY@bf=\textbf\def\PY@tc##1{\textcolor[rgb]{0.00,0.50,0.00}{##1}}}
\@namedef{PY@tok@kr}{\let\PY@bf=\textbf\def\PY@tc##1{\textcolor[rgb]{0.00,0.50,0.00}{##1}}}
\@namedef{PY@tok@bp}{\def\PY@tc##1{\textcolor[rgb]{0.00,0.50,0.00}{##1}}}
\@namedef{PY@tok@fm}{\def\PY@tc##1{\textcolor[rgb]{0.00,0.00,1.00}{##1}}}
\@namedef{PY@tok@vc}{\def\PY@tc##1{\textcolor[rgb]{0.10,0.09,0.49}{##1}}}
\@namedef{PY@tok@vg}{\def\PY@tc##1{\textcolor[rgb]{0.10,0.09,0.49}{##1}}}
\@namedef{PY@tok@vi}{\def\PY@tc##1{\textcolor[rgb]{0.10,0.09,0.49}{##1}}}
\@namedef{PY@tok@vm}{\def\PY@tc##1{\textcolor[rgb]{0.10,0.09,0.49}{##1}}}
\@namedef{PY@tok@sa}{\def\PY@tc##1{\textcolor[rgb]{0.73,0.13,0.13}{##1}}}
\@namedef{PY@tok@sb}{\def\PY@tc##1{\textcolor[rgb]{0.73,0.13,0.13}{##1}}}
\@namedef{PY@tok@sc}{\def\PY@tc##1{\textcolor[rgb]{0.73,0.13,0.13}{##1}}}
\@namedef{PY@tok@dl}{\def\PY@tc##1{\textcolor[rgb]{0.73,0.13,0.13}{##1}}}
\@namedef{PY@tok@s2}{\def\PY@tc##1{\textcolor[rgb]{0.73,0.13,0.13}{##1}}}
\@namedef{PY@tok@sh}{\def\PY@tc##1{\textcolor[rgb]{0.73,0.13,0.13}{##1}}}
\@namedef{PY@tok@s1}{\def\PY@tc##1{\textcolor[rgb]{0.73,0.13,0.13}{##1}}}
\@namedef{PY@tok@mb}{\def\PY@tc##1{\textcolor[rgb]{0.40,0.40,0.40}{##1}}}
\@namedef{PY@tok@mf}{\def\PY@tc##1{\textcolor[rgb]{0.40,0.40,0.40}{##1}}}
\@namedef{PY@tok@mh}{\def\PY@tc##1{\textcolor[rgb]{0.40,0.40,0.40}{##1}}}
\@namedef{PY@tok@mi}{\def\PY@tc##1{\textcolor[rgb]{0.40,0.40,0.40}{##1}}}
\@namedef{PY@tok@il}{\def\PY@tc##1{\textcolor[rgb]{0.40,0.40,0.40}{##1}}}
\@namedef{PY@tok@mo}{\def\PY@tc##1{\textcolor[rgb]{0.40,0.40,0.40}{##1}}}
\@namedef{PY@tok@ch}{\let\PY@it=\textit\def\PY@tc##1{\textcolor[rgb]{0.24,0.48,0.48}{##1}}}
\@namedef{PY@tok@cm}{\let\PY@it=\textit\def\PY@tc##1{\textcolor[rgb]{0.24,0.48,0.48}{##1}}}
\@namedef{PY@tok@cpf}{\let\PY@it=\textit\def\PY@tc##1{\textcolor[rgb]{0.24,0.48,0.48}{##1}}}
\@namedef{PY@tok@c1}{\let\PY@it=\textit\def\PY@tc##1{\textcolor[rgb]{0.24,0.48,0.48}{##1}}}
\@namedef{PY@tok@cs}{\let\PY@it=\textit\def\PY@tc##1{\textcolor[rgb]{0.24,0.48,0.48}{##1}}}

\def\PYZbs{\char`\\}
\def\PYZus{\char`\_}
\def\PYZob{\char`\{}
\def\PYZcb{\char`\}}
\def\PYZca{\char`\^}
\def\PYZam{\char`\&}
\def\PYZlt{\char`\<}
\def\PYZgt{\char`\>}
\def\PYZsh{\char`\#}
\def\PYZpc{\char`\%}
\def\PYZdl{\char`\$}
\def\PYZhy{\char`\-}
\def\PYZsq{\char`\'}
\def\PYZdq{\char`\"}
\def\PYZti{\char`\~}
% for compatibility with earlier versions
\def\PYZat{@}
\def\PYZlb{[}
\def\PYZrb{]}
\makeatother


    % For linebreaks inside Verbatim environment from package fancyvrb.
    \makeatletter
        \newbox\Wrappedcontinuationbox
        \newbox\Wrappedvisiblespacebox
        \newcommand*\Wrappedvisiblespace {\textcolor{red}{\textvisiblespace}}
        \newcommand*\Wrappedcontinuationsymbol {\textcolor{red}{\llap{\tiny$\m@th\hookrightarrow$}}}
        \newcommand*\Wrappedcontinuationindent {3ex }
        \newcommand*\Wrappedafterbreak {\kern\Wrappedcontinuationindent\copy\Wrappedcontinuationbox}
        % Take advantage of the already applied Pygments mark-up to insert
        % potential linebreaks for TeX processing.
        %        {, <, #, %, $, ' and ": go to next line.
        %        _, }, ^, &, >, - and ~: stay at end of broken line.
        % Use of \textquotesingle for straight quote.
        \newcommand*\Wrappedbreaksatspecials {%
            \def\PYGZus{\discretionary{\char`\_}{\Wrappedafterbreak}{\char`\_}}%
            \def\PYGZob{\discretionary{}{\Wrappedafterbreak\char`\{}{\char`\{}}%
            \def\PYGZcb{\discretionary{\char`\}}{\Wrappedafterbreak}{\char`\}}}%
            \def\PYGZca{\discretionary{\char`\^}{\Wrappedafterbreak}{\char`\^}}%
            \def\PYGZam{\discretionary{\char`\&}{\Wrappedafterbreak}{\char`\&}}%
            \def\PYGZlt{\discretionary{}{\Wrappedafterbreak\char`\<}{\char`\<}}%
            \def\PYGZgt{\discretionary{\char`\>}{\Wrappedafterbreak}{\char`\>}}%
            \def\PYGZsh{\discretionary{}{\Wrappedafterbreak\char`\#}{\char`\#}}%
            \def\PYGZpc{\discretionary{}{\Wrappedafterbreak\char`\%}{\char`\%}}%
            \def\PYGZdl{\discretionary{}{\Wrappedafterbreak\char`\$}{\char`\$}}%
            \def\PYGZhy{\discretionary{\char`\-}{\Wrappedafterbreak}{\char`\-}}%
            \def\PYGZsq{\discretionary{}{\Wrappedafterbreak\textquotesingle}{\textquotesingle}}%
            \def\PYGZdq{\discretionary{}{\Wrappedafterbreak\char`\"}{\char`\"}}%
            \def\PYGZti{\discretionary{\char`\~}{\Wrappedafterbreak}{\char`\~}}%
        }
        % Some characters . , ; ? ! / are not pygmentized.
        % This macro makes them "active" and they will insert potential linebreaks
        \newcommand*\Wrappedbreaksatpunct {%
            \lccode`\~`\.\lowercase{\def~}{\discretionary{\hbox{\char`\.}}{\Wrappedafterbreak}{\hbox{\char`\.}}}%
            \lccode`\~`\,\lowercase{\def~}{\discretionary{\hbox{\char`\,}}{\Wrappedafterbreak}{\hbox{\char`\,}}}%
            \lccode`\~`\;\lowercase{\def~}{\discretionary{\hbox{\char`\;}}{\Wrappedafterbreak}{\hbox{\char`\;}}}%
            \lccode`\~`\:\lowercase{\def~}{\discretionary{\hbox{\char`\:}}{\Wrappedafterbreak}{\hbox{\char`\:}}}%
            \lccode`\~`\?\lowercase{\def~}{\discretionary{\hbox{\char`\?}}{\Wrappedafterbreak}{\hbox{\char`\?}}}%
            \lccode`\~`\!\lowercase{\def~}{\discretionary{\hbox{\char`\!}}{\Wrappedafterbreak}{\hbox{\char`\!}}}%
            \lccode`\~`\/\lowercase{\def~}{\discretionary{\hbox{\char`\/}}{\Wrappedafterbreak}{\hbox{\char`\/}}}%
            \catcode`\.\active
            \catcode`\,\active
            \catcode`\;\active
            \catcode`\:\active
            \catcode`\?\active
            \catcode`\!\active
            \catcode`\/\active
            \lccode`\~`\~
        }
    \makeatother

    \let\OriginalVerbatim=\Verbatim
    \makeatletter
    \renewcommand{\Verbatim}[1][1]{%
        %\parskip\z@skip
        \sbox\Wrappedcontinuationbox {\Wrappedcontinuationsymbol}%
        \sbox\Wrappedvisiblespacebox {\FV@SetupFont\Wrappedvisiblespace}%
        \def\FancyVerbFormatLine ##1{\hsize\linewidth
            \vtop{\raggedright\hyphenpenalty\z@\exhyphenpenalty\z@
                \doublehyphendemerits\z@\finalhyphendemerits\z@
                \strut ##1\strut}%
        }%
        % If the linebreak is at a space, the latter will be displayed as visible
        % space at end of first line, and a continuation symbol starts next line.
        % Stretch/shrink are however usually zero for typewriter font.
        \def\FV@Space {%
            \nobreak\hskip\z@ plus\fontdimen3\font minus\fontdimen4\font
            \discretionary{\copy\Wrappedvisiblespacebox}{\Wrappedafterbreak}
            {\kern\fontdimen2\font}%
        }%

        % Allow breaks at special characters using \PYG... macros.
        \Wrappedbreaksatspecials
        % Breaks at punctuation characters . , ; ? ! and / need catcode=\active
        \OriginalVerbatim[#1,codes*=\Wrappedbreaksatpunct]%
    }
    \makeatother

    % Exact colors from NB
    \definecolor{incolor}{HTML}{303F9F}
    \definecolor{outcolor}{HTML}{D84315}
    \definecolor{cellborder}{HTML}{CFCFCF}
    \definecolor{cellbackground}{HTML}{F7F7F7}

    % prompt
    \makeatletter
    \newcommand{\boxspacing}{\kern\kvtcb@left@rule\kern\kvtcb@boxsep}
    \makeatother
    \newcommand{\prompt}[4]{
        {\ttfamily\llap{{\color{#2}[#3]:\hspace{3pt}#4}}\vspace{-\baselineskip}}
    }
    

    
    % Prevent overflowing lines due to hard-to-break entities
    \sloppy
    % Setup hyperref package
    \hypersetup{
      breaklinks=true,  % so long urls are correctly broken across lines
      colorlinks=true,
      urlcolor=urlcolor,
      linkcolor=linkcolor,
      citecolor=citecolor,
      }
    % Slightly bigger margins than the latex defaults
    
    \geometry{verbose,tmargin=1in,bmargin=1in,lmargin=1in,rmargin=1in}
    
    

\begin{document}
    
    \maketitle
    
    

    
    \begin{tcolorbox}[breakable, size=fbox, boxrule=1pt, pad at break*=1mm,colback=cellbackground, colframe=cellborder]
\prompt{In}{incolor}{17}{\boxspacing}
\begin{Verbatim}[commandchars=\\\{\}]
\PY{k+kn}{from} \PY{n+nn}{Bio} \PY{k+kn}{import} \PY{n}{SeqIO}
\PY{k+kn}{from} \PY{n+nn}{Bio}\PY{n+nn}{.}\PY{n+nn}{Align} \PY{k+kn}{import} \PY{n}{MultipleSeqAlignment}
\PY{k+kn}{from} \PY{n+nn}{Bio}\PY{n+nn}{.}\PY{n+nn}{SeqRecord} \PY{k+kn}{import} \PY{n}{SeqRecord}
\PY{k+kn}{from} \PY{n+nn}{Bio}\PY{n+nn}{.}\PY{n+nn}{Seq} \PY{k+kn}{import} \PY{n}{Seq}
\PY{k+kn}{import} \PY{n+nn}{random}
\end{Verbatim}
\end{tcolorbox}

    \subsection{reading Input.fasta}\label{reading-input.fasta}

    \begin{tcolorbox}[breakable, size=fbox, boxrule=1pt, pad at break*=1mm,colback=cellbackground, colframe=cellborder]
\prompt{In}{incolor}{18}{\boxspacing}
\begin{Verbatim}[commandchars=\\\{\}]
\PY{c+c1}{\PYZsh{} Path to the input.fasta file}
\PY{n}{fasta\PYZus{}file} \PY{o}{=} \PY{l+s+s2}{\PYZdq{}}\PY{l+s+s2}{input.fasta}\PY{l+s+s2}{\PYZdq{}}

\PY{c+c1}{\PYZsh{} Read all sequences from the fasta file}
\PY{n}{sequences} \PY{o}{=} \PY{n+nb}{list}\PY{p}{(}\PY{n}{SeqIO}\PY{o}{.}\PY{n}{parse}\PY{p}{(}\PY{n}{fasta\PYZus{}file}\PY{p}{,} \PY{l+s+s2}{\PYZdq{}}\PY{l+s+s2}{fasta}\PY{l+s+s2}{\PYZdq{}}\PY{p}{)}\PY{p}{)}

\PY{c+c1}{\PYZsh{} Print the sequences}
\PY{k}{for} \PY{n}{seq} \PY{o+ow}{in} \PY{n}{sequences}\PY{p}{:}
    \PY{n+nb}{print}\PY{p}{(}\PY{n}{seq}\PY{o}{.}\PY{n}{id}\PY{p}{)}
    \PY{n+nb}{print}\PY{p}{(}\PY{n}{seq}\PY{o}{.}\PY{n}{seq}\PY{p}{)}
\end{Verbatim}
\end{tcolorbox}

    \begin{Verbatim}[commandchars=\\\{\}]
MINE01
AGGAACAGGATGAACAGTTTATCCCCCTCTTTCAAGAAATATTTCTCATAGAGGAGCTTCAGTTGACTTAAGAATCTTCA
GATTACATTTAGCAGGATTCTCTTCAATCCTAGGAGCTATCAACTTCATTACAACTATTTTAAATATACGACCAAAAAAC
TTAAATCTAGAATTAATCCCTTTATTTGTATGATCTGTTCTAATCACAGCAATTTTATTATTACTTAGATTACCAGTCTT
AGCAGCAGCTATTACCATACTTCTAACAGATCGTAATCTCAATACCTCATTTTTCGACCCTGCAGGTGGAGGAGACCCAA
TCCTTTACCAACACTTATTCTGATTTTTTGGTCACCCTGAAGTTTACATTCTTATTCTACCAGGATTTGGTATTATTTCT
CACATTATCTCATTCCATAGAGGAAAAAAAAATCCTTTCGGAAATTTAGGTATAATCTATGCTATATTAACTATTGGAAT
TTTAGGATTCATTGTATGAGCTCATCACATATTCACTATTGGTCTTGATATTGATACT
MINE03
AGGAACAGGATGAACAGTTTATCCCCCTCTTTCAAGAAATATTTCTCATAGAGGAGCTTCAGTTGACTTAAGAATCTTCA
GATTACATTTAGCAGGATTCTCTTCAATCCTAGGAGCTATCAACTTCATTACAACTATTTTAAATATACGACCAAAAAAC
TTAAATCTAGAATTAATCCCTTTATTTGTATGATCTGTTCTAATCACAGCAATTTTATTATTACTTAGATTACCAGTCTT
AGCAGCAGCTATTACCATACTTCTAACAGATCGTAATCTCAATACCTCATTTTTCGACCCTGCAGGTGGAGGAGACCCAA
TCCTTTACCAACACTTATTCTGATTTTTTGGTCACCCTGAAGTTTACATTCTTATTCTACCAGGATTTGGTATTATTTCT
CACATTATCTCATTCCATAGAGGAAAAAAAAATCCTTTCGGAAATTTAGGTATAATCTATGCTATATTAACTATTGGAAT
TTTAGGATTCATTGTATGAGCTCATCACATATTCACTATTGGTCTTGATATTGATACT
MINE04
AGGAACAGGATGAACAGTTTATCCCCCTCTTTCAAGAAATATTTCTCATAGAGGAGCTTCAGTTGACTTAAGAATCTTCA
GATTACATTTAGCAGGATTCTCTTCAATCCTAGGAGCTATCAACTTCATTACAACTATTTTAAATATACGACCAAAAAAC
TTAAATCTAGAATTAATCCCTTTATTTGTATGATCTGTTCTAATCACAGCAATTTTATTATTACTTAGATTACCAGTCTT
AGCAGCAGCTATTACCATACTTCTAACAGATCGTAATCTCAATACCTCATTTTTCGACCCTGCAGGTGGAGGAGACCCAA
TCCTTTACCAACACTTATTCTGATTTTTTGGTCACCCTGAAGTTTACATTCTTATTCTACCAGGATTTGGTATTATTTCT
CACATTATCTCATTCCATAGAGGAAAAAAAAATCCTTTCGGAAATTTAGGTATAATCTATGCTATATTAACTATTGGAAT
TTTAGGATTCATTGTATGAGCTCATCACATATTCACTATTGGTCTTGATATTGATACT
MINE05
AGGAACAGGATGAACAGTTTATCCCCCTCTTTCAAGAAATATTTCTCATAGAGGAGCTTCAGTTGACTTAAGAATCTTCA
GATTACATTTAGCAGGATTCTCTTCAATCCTAGGAGCTATCAACTTCATTACAACTATTTTAAATATACGACCAAAAAAC
TTAAATCTAGAATTAATCCCTTTATTTGTATGATCTGTTCTAATCACAGCAATTTTATTATTACTTAGATTACCAGTCTT
AGCAGCAGCTATTACCATACTTCTAACAGATCGTAATCTCAATACCTCATTTTTCGACCCTGCAGGTGGAGGAGACCCAA
TCCTTTACCAACACTTATTCTGATTTTTTGGTCACCCTGAAGTTTACATTCTTATTCTACCAGGATTTGGTATTATTTCT
CACATTATCTCATTCCATAGAGGAAAAAAAAATCCTTTCGGAAATTTAGGTATAATCTATGCTATATTAACTATTGGAAT
TTTAGGATTCATTGTATGAGCTCATCACATATTCACTATTGGTCTTGATATTGATACT
MINE06
AGGAACAGGATGAACAGTTTATCCCCCTCTTTCAAGAAATATTTCTCATAGAGGAGCTTCAGTTGACTTAAGAATCTTCA
GATTACATTTAGCAGGATTCTCTTCAATCCTAGGAGCTATCAACTTCATTACAACTATTTTAAATATACGACCAAAAAAC
TTAAATCTAGAATTAATCCCTTTATTTGTATGATCTGTTCTAATCACAGCAATTTTATTATTACTTAGATTACCAGTCTT
AGCAGCAGCTATTACCATACTTCTAACAGATCGTAATCTCAATACCTCATTTTTCGACCCTGCAGGTGGAGGAGACCCAA
TCCTTTACCAACACTTATTCTGATTTTTTGGTCACCCTGAAGTTTACATTCTTATTCTACCAGGATTTGGTATTATTTCT
CACATTATCTCATTCCATAGAGGAAAAAAAAATCCTTTCGGAAATTTAGGTATAATCTATGCTATATTAACTATTGGAAT
TTTAGGATTCATTGTATGAGCTCATCACATATTCACTATTGGTCTTGATATTGATACT
KO-A1
AGGAACAGGATGAACAGTTTATCCCCCTCTTTCAAGAAATATTTCTCATAGAGGAGCTTCAGTTGACTTAAGAATCTTCA
GATTACATTTAGCAGGATTCTCTTCAATCCTAGGAGCTATCAACTTCATTACAACTATTTTAAATATACGACCAAAAAAC
TTAAATCTAGAATTAATCCCTTTATTTGTATGATCTGTTCTAATCACAGCAATTTTATTATTACTTAGATTACCAGTCTT
AGCAGCAGCTATTACCATACTTCTAACAGATCGTAATCTCAATACCTCATTTTTCGACCCTGCAGGTGGAGGAGACCCAA
TCCTTTACCAACACTTATTCTGATTTTTTGGTCACCCTGAAGTTTACATTCTTATTCTACCAGGATTTGGTATTATTTCT
CACATTATCTCATTCCATAGAGGAAAAAAAAATCCTTTCGGAAATTTAGGTATAATCTATGCTATATTAACTATTGGAAT
TTTAGGATTCATTGTATGAGCTCATCACATATTCACTATTGGTCTTGATATTGATACT
KO-A2
AGGAACAGGATGAACAGTTTATCCCCCTCTTTCAAGAAATATTTCTCATAGAGGAGCTTCAGTTGACTTAAGAATCTTCA
GATTACATTTAGCAGGATTCTCTTCAATCCTAGGAGCTATCAACTTCATTACAACTATTTTAAATATACGACCAAAAAAC
TTAAATCTAGAATTAATCCCTTTATTTGTATGATCTGTTCTAATCACAGCAATTTTATTATTACTTAGATTACCAGTCTT
AGCAGCAGCTATTACCATACTTCTAACAGATCGTAATCTCAATACCTCATTTTTCGACCCTGCAGGTGGAGGAGACCCAA
TCCTTTACCAACACTTATTCTGATTTTTTGGTCACCCTGAAGTTTACATTCTTATTCTACCAGGATTTGGTATTATTTCT
CACATTATCTCATTCCATAGAGGAAAAAAAAATCCTTTCGGAAATTTAGGTATAATCTATGCTATATTAACTATTGGAAT
TTTAGGATTCATTGTATGAGCTCATCACATATTCACTATTGGTCTTGATATTGATACT
KO-A3
AGGAACAGGATGAACAGTTTATCCCCCTCTTTCAAGAAATATTTCTCATAGAGGAGCTTCAGTTGACTTAAGAATCTTCA
GATTACATTTAGCAGGATTCTCTTCAATCCTAGGAGCTATCAACTTCATTACAACTATTTTAAATATACGACCAAAAAAC
TTAAATCTAGAATTAATCCCTTTATTTGTATGATCTGTTCTAATCACAGCAATTTTATTATTACTTAGATTACCAGTCTT
AGCAGCAGCTATTACCATACTTCTAACAGATCGTAATATCAATACCTCATTTTTCGACCCTGCAGGTGGAGGAGACCCAA
TCCTTTACCAACACTTATTCTGATTTTTTGGTCACCCTGAAGTTTACATTCTTATTCTACCAGGATTTGGTATTATTTCT
CACATTATCTCATTCCATAGAGGAAAAAAAAATCCTTTCGGAAATTTAGGTATAATCTATGCTATATTAACTATTGGAAT
TTTAGGATTCATTGTATGAGCTCATCACATATTCACTATTGGTCTTGATATTGATACT
KO-A4
AGGAACAGGATGAACAGTTTATCCCCCTCTTTCAAGAAATATTTCTCATAGAGGAGCTTCAGTTGACTTAAGAATCTTCA
GATTACATTTAGCAGGATTCTCTTCAATCCTAGGAGCTATCAACTTCATTACAACTATTTTAAATATACGACCAAAAAAC
TTAAATCTAGAATTAATCCCTTTATTTGTATGATCTGTTCTAATCACAGCAATTTTATTATTACTTAGATTACCAGTCTT
AGCAGCAGCTATTACCATACTTCTAACAGATCGTAATCTCAATACCTCATTTTTCGACCCTGCAGGTGGAGGAGACCCAA
TCCTTTACCAACACTTATTCTGATTTTTTGGTCACCCTGAAGTTTACATTCTTATTCTACCAGGATTTGGTATTATTTCT
CACATTATCTCATTCCATAGAGGAAAAAAAAATCCTTTCGGAAATTTAGGTATAATCTATGCTATATTAACTATTGGAAT
TTTAGGATTCATTGTATGAGCTCATCACATATTCACTATTGGTCTTGATATTGATACT
KO-A5
AGGAACAGGATGAACAGTTTATCCCCCTCTTTCAAGAAATATTTCTCATAGAGGAGCTTCAGTTGACTTAAGAATCTTCA
GATTACATTTAGCAGGATTCTCTTCAATCCTAGGAGCTATCAACTTCATTACAACTATTTTAAATATACGACCAAAAAAC
TTAAATCTAGAATTAATCCCTTTATTTGTATGATCTGTTCTAATCACAGCAATTTTATTATTACTTAGATTACCAGTCTT
AGCAGCAGCTATTACTATACTTCTAACAGATCGTAATCTCAATACCTCATTTTTCGACCCTGCAGGTGGAGGAGACCCAA
TCCTTTACCAACACTTATTCTGATTTTTTGGTCACCCTGAAGTTTACATTCTTATTCTACCAGGATTTGGTATTATTTCT
CACATTATCTCATTCCATAGAGGAAAAAAAAATCCTTTCGGAAATTTAGGTATAATCTATGCTATATTAACTATTGGAAT
TTTAGGATTCATTGTATGAGCTCATCACATATTCACTATTGGTCTTGATATTGATACT
KO-A6
AGGAACAGGATGAACAGTTTATCCCCCTCTTTCAAGAAATATTTCTCATAGAGGAGCTTCAGTTGACTTAAGAATCTTCA
GATTACATTTAGCAGGATTCTCTTCAATCCTAGGAGCTATCAACTTCATTACAACTATTTTAAATATACGACCAAAAAAC
TTAAATCTAGAATTAATCCCTTTATTTGTATGATCTGTTCTAATCACAGCAATTTTATTATTACTTAGATTACCAGTCTT
AGCAGCAGCTATTACCATACTTCTAACAGATCGTAATCTCAATACCTCATTTTTCGACCCTGCAGGTGGAGGAGACCCAA
TCCTTTACCAACACTTATTCTGATTTTTTGGTCACCCTGAAGTTTACATTCTTATTCTACCAGGATTTGGTATTATTTCT
CACATTATCTCATTCCATAGAGGAAAAAAAAATCCTTTCGGAAATTTAGGTATAATCTATGCTATATTAACTATTGGAAT
TTTAGGATTCATTGTATGAGCTCATCACATATTCACTATTGGTCTTGATATTGATACT
KO-A7
AGGAACAGGATGAACAGTTTATCCCCCTCTTTCAAGAAATATTTCTCATAGAGGAGCTTCAGTTGACTTAAGAATCTTCA
GATTACATTTAGCAGGATTCTCTTCAATCCTAGGAGCTATCAACTTCATTACAACTATTTTAAATATACGACCAAAAAAC
TTAAATCTAGAATTAATCCCTTTATTTGTATGATCTGTTCTAATCACAGCAATTTTATTATTACTTAGATTACCAGTCTT
AGCAGCAGCTATTACCATACTTCTAACAGATCGTAATCTCAATACCTCATTTTTCGACCCTGCAGGTGGAGGAGACCCAA
TCCTTTACCAACACTTATTCTGATTTTTTGGTCACCCTGAAGTTTACATTCTTATTCTACCAGGATTTGGTATTATTTCT
CACATTATCTCATTCCATAGAGGAAAAAAAAATCCTTTCGGAAATTTAGGTATAATCTATGCTATATTAACTATTGGAAT
TTTAGGATTCATTGTATGAGCTCATCACATATTCACTATTGGTCTTGATATTGATACT
KO-A8
AGGAACAGGATGAACAGTTTATCCCCCTCTTTCAAGAAATATTTCTCATAGAGGAGCTTCAGTTGACTTAAGAATCTTCA
GATTACATTTAGCAGGATTCTCTTCAATCCTAGGAGCTATCAACTTCATTACAACTATTTTAAATATACGACCAAAAAAC
TTAAATCTAGAATTAATCCCTTTATTTGTATGATCTGTTCTAATCACAGCAATTTTATTATTACTTAGATTACCAGTCTT
AGCAGCAGCTATTACCATACTTCTAACAGATCGTAATCTCAATACCTCATTTTTCGACCCTGCAGGTGGAGGAGACCCAA
TCCTTTACCAACACTTATTCTGATTTTTTGGTCACCCTGAAGTTTACATTCTTATTCTACCAGGATTTGGTATTATTTCT
CACATTATCTCATTCCATAGAGGAAAAAAAAATCCTTTCGGAAATTTAGGTATAATCTATGCTATATTAACTATTGGAAT
TTTAGGATTCATTGTATGAGCTCATCACATATTCACTATTGGTCTTGATATTGATACT
KO-A9
AGGAACAGGATGAACAGTTTATCCCCCTCTTTCAAGAAATATTTCTCATAGAGGAGCTTCAGTTGACTTAAGAATCTTCA
GATTACATTTAGCAGGATTCTCTTCAATCCTAGGAGCTATCAACTTCATTACAACTATTTTAAATATACGACCAAAAAAC
TTAAATCTAGAATTAATCCCTTTATTTGTATGATCTGTTCTAATCACAGCAATTTTATTATTACTTAGATTACCAGTCTT
AGCAGCAGCTATTACCATACTTCTAACAGATCGTAATCTCAATACCTCATTTTTCGACCCTGCAGGTGGAGGAGACCCAA
TCCTTTACCAACACTTATTCTGATTTTTTGGTCACCCTGAAGTTTACATTCTTATTCTACCAGGATTTGGTATTATTTCT
CACATTATCTCATTCCATAGAGGAAAAAAAAATCCTTTCGGAAATTTAGGTATAATCTATGCTATATTAACTATTGGAAT
TTTAGGATTCATTGTATGAGCTCATCACATATTCACTATTGGTCTTGATATTGATACT
KO-A10
AGGAACAGGATGAACAGTTTATCCCCCTCTTTCAAGAAATATTTCTCATAGAGGAGCTTCAGTTGACTTAAGAATCTTCA
GATTACATTTAGCAGGATTCTCTTCAATCCTAGGAGCTATCAACTTCATTACAACTATTTTAAATATACGACCAAAAAAC
TTAAATCTAGAATTAATCCCTTTATTTGTATGATCTGTTCTAATCACAGCAATTTTATTATTACTTAGATTACCAGTCTT
AGCAGCAGCTATTACCATACTTCTAACAGATCGTAATCTCAATACCTCATTTTTCGACCCTGCAGGTGGAGGAGACCCAA
TCCTTTACCAACACTTATTCTGATTTTTTGGTCACCCTGAAGTTTACATTCTTATTCTACCAGGATTTGGTATTATTTCT
CACATTATCTCATTCCATAGAGGAAAAAAAAATCCTTTCGGAAATTTAGGTATAATCTATGCTATATTAACTATTGGAAT
TTTAGGATTCATTGTATGAGCTCATCACATATTCACTATTGGTCTTGATATTGATACT
KO-B1
AGGAACAGGATGAACAGTTTATCCCCCTCTTTCAAGAAATATTTCTCATAGAGGAGCTTCAGTTGACTTAAGAATCTTCA
GATTACATTTAGCAGGATTCTCTTCAATCCTAGGAGCTATCAACTTCATTACAACTATTTTAAATATACGACCAAAAAAC
TTAAATCTAGAATTAATCCCTTTATTTGTATGATCTGTTCTAATCACAGCAATTTTATTATTACTTAGATTACCAGTCTT
AGCAGCAGCTATTACCATACTTCTAACAGATCGTAATCTCAATACCTCATTTTTCGACCCTGCAGGTGGAGGAGACCCAA
TCCTTTACCAACACTTATTCTGATTTTTTGGTCACCCTGAAGTTTACATTCTTATTCTACCAGGATTTGGTATTATTTCT
CACATTATCTCATTCCATAGAGGAAAAAAAAATCCTTTCGGAAATTTAGGTATAATCTATGCTATATTAACTATTGGAAT
TTTAGGATTCATTGTATGAGCTCATCACATATTCACTATTGGTCTTGATATTGATACT
KO-B2
AGGAACAGGATGAACAGTTTATCCCCCTCTTTCAAGAAATATTTCTCATAGAGGAGCTTCAGTTGACTTAAGAATCTTCA
GATTACATTTAGCAGGATTCTCTTCAATCCTAGGAGCTATCAACTTCATTACAACTATTTTAAATATACGACCAAAAAAC
TTAAATCTAGAATTAATCCCTTTATTTGTATGATCTGTTCTAATCACAGCAATTTTATTATTACTTAGATTACCAGTCTT
AGCAGCAGCTATTACCATACTTCTAACAGATCGTAATCTCAATACCTCATTTTTCGACCCTGCAGGTGGAGGAGACCCAA
TCCTTTACCAACACTTATTCTGATTTTTTGGTCACCCTGAAGTTTACATTCTTATTCTACCAGGATTTGGTATTATTTCT
CACATTATCTCATTCCATAGAGGAAAAAAAAATCCTTTCGGAAATTTAGGTATAATCTATGCTATATTAACTATTGGAAT
TTTAGGATTCATTGTATGAGCTCATCACATATTCACTATTGGTCTTGATATTGATACT
KO-B3
AGGAACAGGATGAACAGTTTATCCCCCTCTTTCAAGAAATATTTCTCATAGAGGAGCTTCAGTTGACTTAAGAATCTTCA
GATTACATTTAGCAGGATTCTCTTCAATCCTAGGAGCTATCAACTTCATTACAACTATTTTAAATATACGACCAAAAAAC
TTAAATCTAGAATTAATCCCTTTATTTGTATGATCTGTTCTAATCACAGCAATTTTATTATTACTTAGATTACCAGTCTT
AGCAGCAGCTATTACCATACTTCTAACAGATCGTAATCTCAATACCTCATTTTTCGACCCTGCAGGTGGAGGAGACCCAA
TCCTTTACCAACACTTATTCTGATTTTTTGGTCACCCTGAAGTTTACATTCTTATTCTACCAGGATTTGGTATTATTTCT
CACATTATCTCATTCCATAGAGGAAAAAAAAATCCTTTCGGAAATTTAGGTATAATCTATGCTATATTAACTATTGGAAT
TTTAGGATTCATTGTATGAGCTCATCACATATTCACTATTGGTCTTGATATTGATACT
KO-B4
AGGAACAGGATGAACAGTTTATCCCCCTCTTTCAAGAAATATTTCTCATAGAGGAGCTTCAGTTGACTTAAGAATCTTCA
GATTACATTTAGCAGGATTCTCTTCAATCCTAGGAGCTATCAACTTCATTACAACTATTTTAAATATACGACCAAAAAAC
TTAAATCTAGAATTAATCCCTTTATTTGTATGATCTGTTCTAATCACAGCAATTTTATTATTACTTAGATTACCAGTCTT
AGCAGCAGCTATTACCATACTTCTAACAGATCGTAATCTCAATACCTCATTTTTCGACCCTGCAGGTGGAGGAGACCCAA
TCCTTTACCAACACTTATTCTGATTTTTTGGTCACCCTGAAGTTTACATTCTTATTCTACCAGGATTTGGTATTATTTCT
CACATTATCTCATTCCATAGAGGAAAAAAAAATCCTTTCGGAAATTTAGGTATAATCTATGCTATATTAACTATTGGAAT
TTTAGGATTCATTGTATGAGCTCATCACATATTCACTATTGGTCTTGATATTGATACT
KO-B5
AGGAACAGGATGAACAGTTTATCCCCCTCTTTCAAGAAATATTTCTCATAGAGGAGCTTCAGTTGACTTAAGAATCTTCA
GATTACATTTAGCAGGATTCTCTTCAATCCTAGGAGCTATCAACTTCATTACAACTATTTTAAATATACGACCAAAAAAC
TTAAATCTAGAATTAATCCCTTTATTTGTATGATCTGTTCTAATCACAGCAATTTTATTATTACTTAGATTACCAGTCTT
AGCAGCAGCTATTACCATACTTCTAACAGATCGTAATCTCAATACCTCATTTTTCGACCCTGCAGGTGGAGGAGACCCAA
TCCTTTACCAACACTTATTCTGATTTTTTGGTCACCCTGAAGTTTACATTCTTATTCTACCAGGATTTGGTATTATTTCT
CACATTATCTCATTCCATAGAGGAAAAAAAAATCCTTTCGGAAATTTAGGTATAATCTATGCTATATTAACTATTGGAAT
TTTAGGATTCATTGTATGAGCTCATCACATATTCACTATTGGTCTTGATATTGATACT
KO-B6
AGGAACAGGATGAACAGTTTATCCCCCTCTTTCAAGAAATATTTCTCATAGAGGAGCTTCAGTTGACTTAAGAATCTTCA
GATTACATTTAGCAGGATTCTCTTCAATCCTAGGAGCTATCAACTTCATTACAACTATTTTAAATATACGACCAAAAAAC
TTAAATCTAGAATTAATCCCTTTATTTGTATGATCTGTTCTAATCACAGCAATTTTATTATTACTTAGATTACCAGTCTT
AGCAGCAGCTATTACCATACTTCTAACAGATCGTAATCTCAATACCTCATTTTTCGACCCTGCAGGTGGAGGAGACCCAA
TCCTTTACCAACACTTATTCTGATTTTTTGGTCACCCTGAAGTTTACATTCTTATTCTACCAGGATTTGGTATTATTTCT
CACATTATCTCATTCCATAGAGGAAAAAAAAATCCTTTCGGAAATTTAGGTATAATCTATGCTATATTAACTATTGGAAT
TTTAGGATTCATTGTATGAGCTCATCACATATTCACTATTGGTCTTGATATTGATACT
KO-B7
AGGAACAGGATGAACAGTTTATCCCCCTCTTTCAAGAAATATTTCTCATAGAGGAGCTTCAGTTGACTTAAGAATCTTCA
GATTACATTTAGCAGGATTCTCTTCAATCCTAGGAGCTATCAACTTCATTACAACTATTTTAAATATACGACCAAAAAAC
TTAAATCTAGAATTAATCCCTTTATTTGTATGATCTGTTCTAATCACAGCAATTTTATTATTACTTAGATTACCAGTCTT
AGCAGCAGCTATTACCATACTTCTAACAGATCGTAATCTCAATACCTCATTTTTCGACCCTGCAGGTGGAGGAGACCCAA
TCCTTTACCAACACTTATTCTGATTTTTTGGTCACCCTGAAGTTTACATTCTTATTCTACCAGGATTTGGTATTATTTCT
CACATTATCTCATTCCATAGAGGAAAAAAAAATCCTTTCGGAAATTTAGGTATAATCTATGCTATATTAACTATTGGAAT
TTTAGGATTCATTGTATGAGCTCATCACATATTCACTATTGGTCTTGATATTGATACT
KO-B8
AGGAACAGGATGAACAGTTTATCCCCCTCTTTCAAGAAATATTTCTCATAGAGGAGCTTCAGTTGACTTAAGAATCTTCA
GATTACATTTAGCAGGATTCTCTTCAATCCTAGGAGCTATCAACTTCATTACAACTATTTTAAATATACGACCAAAAAAC
TTAAATCTAGAATTAATCCCTTTATTTGTATGATCTGTTCTAATCACAGCAATTTTATTATTACTTAGATTACCAGTCTT
AGCAGCAGCTATTACCATACTTCTAACAGATCGTAATCTCAATACCTCATTTTTCGACCCTGCAGGTGGAGGAGACCCAA
TCCTTTACCAACACTTATTCTGATTTTTTGGTCACCCTGAAGTTTACATTCTTATTCTACCAGGATTTGGTATTATTTCT
CACATTATCTCATTCCATAGAGGAAAAAAAAATCCTTTCGGAAATTTAGGTATAATCTATGCTATATTAACTATTGGAAT
TTTAGGATTCATTGTATGAGCTCATCACATATTCACTATTGGTCTTGATATTGATACT
KO-B9
AGGAACAGGATGAACAGTTTATCCCCCTCTTTCAAGAAATATTTCTCATAGAGGAGCTTCAGTTGACTTAAGAATCTTCA
GATTACATTTAGCAGGATTCTCTTCAATCCTAGGAGCTATCAACTTCATTACAACTATTTTAAATATACGACCAAAAAAC
TTAAATCTAGAATTAATCCCTTTATTTGTATGATCTGTTCTAATCACAGCAATTTTATTATTACTTAGATTACCAGTCTT
AGCAGCAGCTATTACCATACTTCTAACAGATCGTAATCTCAATACCTCATTTTTCGACCCTGCAGGTGGAGGAGACCCAA
TCCTTTACCAACACTTATTCTGATTTTTTGGTCACCCTGAAGTTTACATTCTTATTCTACCAGGATTTGGTATTATTTCT
CACATTATCTCATTCCATAGAGGAAAAAAAAATCCTTTCGGAAATTTAGGTATAATCTATGCTATATTAACTATTGGAAT
TTTAGGATTCATTGTATGAGCTCATCACATATTCACTATTGGTCTTGATATTGATACT
KO-B10
AGGAACAGGATGAACAGTTTATCCCCCTCTTTCAAGAAATATTTCTCATAGAGGAGCTTCAGTTGACTTAAGAATCTTCA
GATTACATTTAGCAGGATTCTCTTCAATCCTAGGAGCTATCAACTTCATTACAACTATTTTAAATATACGACCAAAAAAC
TTAAATCTAGAATTAATCCCTTTATTTGTATGATCTGTTCTAATCACAGCAATTTTATTATTACTTAGATTACCAGTCTT
AGCAGCAGCTATTACCATACTTCTAACAGATCGTAATCTCAATACCTCATTTTTCGACCCTGCAGGTGGAGGAGACCCAA
TCCTTTACCAACACTTATTCTGATTTTTTGGTCACCCTGAAGTTTACATTCTTATTCTACCAGGATTTGGTATTATTTCT
CACATTATCTCATTCCATAGAGGAAAAAAAAATCCTTTCGGAAATTTAGGTATAATCTATGCTATATTAACTATTGGAAT
TTTAGGATTCATTGTATGAGCTCATCACATATTCACTATTGGTCTTGATATTGATACT
KO-J1
AGGAACAGGATGAACAGTTTATCCCCCTCTTTCAAGAAATATTTCTCATAGAGGAGCTTCAGTTGACTTAAGAATCTTCA
GATTACATTTAGCAGGATTCTCTTCAATCCTAGGAGCTATCAACTTCATTACAACTATTTTAAATATACGACCAAAAAAC
TTAAATCTAGAATTAATCCCTTTATTTGTATGATCTGTTCTAATCACAGCAATTTTATTATTACTTAGATTACCAGTCTT
AGCAGCAGCTATTACCATACTTCTAACAGATCGTAATCTCAATACCTCATTTTTCGACCCTGCAGGTGGAGGAGACCCAA
TCCTTTACCAACACTTATTCTGATTTTTTGGTCACCCTGAAGTTTACATTCTTATTCTACCAGGATTTGGTATTATTTCT
CACATTATCTCATTCCATAGAGGAAAAAAAAATCCTTTCGGAAATTTAGGTATAATCTATGCTATATTAACTATTGGAAT
TTTAGGATTCATTGTATGAGCTCATCACATATTCACTATTGGTCTTGATATTGATACT
KO-J2
AGGAACAGGATGAACAGTTTATCCCCCTCTTTCAAGAAATATTTCTCATAGAGGAGCTTCAGTTGACTTAAGAATCTTCA
GATTACATTTAGCAGGATTCTCTTCAATCCTAGGAGCTATCAACTTCATTACAACTATTTTAAATATACGACCAAAAAAC
TTAAATCTAGAATTAATCCCTTTATTTGTATGATCTGTTCTAATCACAGCAATTTTATTATTACTTAGATTACCAGTCTT
AGCAGCAGCTATTACCATACTTCTAACAGATCGTAATCTCAATACCTCATTTTTCGACCCTGCAGGTGGAGGAGACCCAA
TCCTTTACCAACACTTATTCTGATTTTTTGGTCACCCTGAAGTTTACATTCTTATTCTACCAGGATTTGGTATTATTTCT
CACATTATCTCATTCCATAGAGGAAAAAAAAATCCTTTCGGAAATTTAGGTATAATCTATGCTATATTAACTATTGGAAT
TTTAGGATTCATTGTATGAGCTCATCACATATTCACTATTGGTCTTGATATTGATACT
KO-J3
AGGAACAGGATGAACAGTTTATCCCCCTCTTTCAAGAAATATTTCTCATAGAGGAGCTTCAGTTGACTTAAGAATCTTCA
GATTACATTTAGCAGGATTCTCTTCAATCCTAGGAGCTATCAACTTCATTACAACTATTTTAAATATACGACCAAAAAAC
TTAAATCTAGAATTAATCCCTTTATTTGTATGATCTGTTCTAATCACAGCAATTTTATTATTACTTAGATTACCAGTCTT
AGCAGCAGCTATTACCATACTTCTAACAGATCGTAATCTCAATACCTCATTTTTCGACCCTGCAGGTGGAGGAGACCCAA
TCCTTTACCAACACTTATTCTGATTTTTTGGTCACCCTGAAGTTTACATTCTTATTCTACCAGGATTTGGTATTATTTCT
CACATTATCTCATTCCATAGAGGAAAAAAAAATCCTTTCGGAAATTTAGGTATAATCTATGCTATATTAACTATTGGAAT
TTTAGGATTCATTGTATGAGCTCATCACATATTCACTATTGGTCTTGATATTGATACT
KO-J4
AGGAACAGGATGAACAGTTTATCCCCCTCTTTCAAGAAATATTTCTCATAGAGGAGCTTCAGTTGACTTAAGAATCTTCA
GATTACATTTAGCAGGATTCTCTTCAATCCTAGGAGCTATCAACTTCATTACAACTATTTTAAATATACGACCAAAAAAC
TTAAATCTAGAATTAATCCCTTTATTTGTATGATCTGTTCTAATCACAGCAATTTTATTATTACTTAGATTACCAGTCTT
AGCAGCAGCTATTACCATACTTCTAACAGATCGTAATCTCAATACCTCATTTTTCGACCCTGCAGGTGGAGGAGACCCAA
TCCTTTACCAACACTTATTCTGATTTTTTGGTCACCCTGAAGTTTACATTCTTATTCTACCAGGATTTGGTATTATTTCT
CACATTATCTCATTCCATAGAGGAAAAAAAAATCCTTTCGGAAATTTAGGTATAATCTATGCTATATTAACTATTGGAAT
TTTAGGATTCATTGTATGAGCTCATCACATATTCACTATTGGTCTTGATATTGATACT
KO-J5
AGGAACAGGATGAACAGTTTATCCCCCTCTTTCAAGAAATATTTCTCATAGAGGAGCTTCAGTTGACTTAAGAATCTTCA
GATTACATTTAGCAGGATTCTCTTCAATCCTAGGAGCTATCAACTTCATTACAACTATTTTAAATATACGACCAAAAAAC
TTAAATCTAGAATTAATCCCTTTATTTGTATGATCTGTTCTAATCACAGCAATTTTATTATTACTTAGATTACCAGTCTT
AGCAGCAGCTATTACCATACTTCTAACAGATCGTAATCTCAATACCTCATTTTTCGACCCTGCAGGTGGAGGAGACCCAA
TCCTTTACCAACACTTATTCTGATTTTTTGGTCACCCTGAAGTTTACATTCTTATTCTACCAGGATTTGGTATTATTTCT
CACATTATCTCATTCCATAGAGGAAAAAAAAATCCTTTCGGAAATTTAGGTATAATCTATGCTATATTAACTATTGGAAT
TTTAGGATTCATTGTATGAGCTCATCACATATTCACTATTGGTCTTGATATTGATACT
KO-J6
AGGAACAGGATGAACAGTTTATCCCCCTCTTTCAAGAAATATTTCTCATAGAGGAGCTTCAGTTGACTTAAGAATCTTCA
GATTACATTTAGCAGGATTCTCTTCAATCCTAGGAGCTATCAACTTCATTACAACTATTTTAAATATACGACCAAAAAAC
TTAAATCTAGAATTAATCCCTTTATTTGTATGATCTGTTCTAATCACAGCAATTTTATTATTACTTAGATTACCAGTCTT
AGCAGCAGCTATTACCATACTTCTAACAGATCGTAATCTCAATACCTCATTTTTCGACCCTGCAGGTGGAGGAGACCCAA
TCCTTTACCAACACTTATTCTGATTTTTTGGTCACCCTGAAGTTTACATTCTTATTCTACCAGGATTTGGTATTATTTCT
CACATTATCTCATTCCATAGAGGAAAAAAAAATCCTTTCGGAAATTTAGGTATAATCTATGCTATATTAACTATTGGAAT
TTTAGGATTCATTGTATGAGCTCATCACATATTCACTATTGGTCTTGATATTGATACT
KO-J7
AGGAACAGGATGAACAGTTTATCCCCCTCTTTCAAGAAATATTTCTCATAGAGGAGCTTCAGTTGACTTAAGAATCTTCA
GATTACATTTAGCAGGATTCTCTTCAATCCTAGGAGCTATCAACTTCATTACAACTATTTTAAATATACGACCAAAAAAC
TTAAATCTAGAATTAATCCCTTTATTTGTATGATCTGTTCTAATCACAGCAATTTTATTATTACTTAGATTACCAGTCTT
AGCAGCAGCTATTACCATACTTCTAACAGATCGTAATCTCAATACCTCATTTTTCGACCCTGCAGGTGGAGGAGACCCAA
TCCTTTACCAACACTTATTCTGATTTTTTGGTCACCCTGAAGTTTACATTCTTATTCTACCAGGATTTGGTATTATTTCT
CACATTATCTCATTCCATAGAGGAAAAAAAAATCCTTTCGGAAATTTAGGTATAATCTATGCTATATTAACTATTGGAAT
TTTAGGATTCATTGTATGAGCTCATCACATATTCACTATTGGTCTTGATATTGATACT
KO-J8
AGGAACAGGATGAACAGTTTATCCCCCTCTTTCAAGAAATATTTCTCATAGAGGAGCTTCAGTTGACTTAAGAATCTTCA
GATTACATTTAGCAGGATTCTCTTCAATCCTAGGAGCTATCAACTTCATTACAACTATTTTAAATATACGACCAAAAAAC
TTAAATCTAGAATTAATCCCTTTATTTGTATGATCTGTTCTAATCACAGCAATTTTATTATTACTTAGATTACCAGTCTT
AGCAGCAGCTATTACCATACTTCTAACAGATCGTAATCTCAATACCTCATTTTTCGACCCTGCAGGTGGAGGAGACCCAA
TCCTTTACCAACACTTATTCTGATTTTTTGGTCACCCTGAAGTTTACATTCTTATTCTACCAGGATTTGGTATTATTTCT
CACATTATCTCATTCCATAGAGGAAAAAAAAATCCTTTCGGAAATTTAGGTATAATCTATGCTATATTAACTATTGGAAT
TTTAGGATTCATTGTATGAGCTCATCACATATTCACTATTGGTCTTGATATTGATACT
KO-J9
AGGAACAGGATGAACAGTTTATCCCCCTCTTTCAAGAAATATTTCTCATAGAGGAGCTTCAGTTGACTTAAGAATCTTCA
GATTACATTTAGCAGGATTCTCTTCAATCCTAGGAGCTATCAACTTCATTACAACTATTTTAAATATACGACCAAAAAAC
TTAAATCTAGAATTAATCCCTTTATTTGTATGATCTGTTCTAATCACAGCAATTTTATTATTACTTAGATTACCAGTCTT
AGCAGCAGCTATTACCATACTTCTAACAGATCGTAATCTCAATACCTCATTTTTCGACCCTGCAGGTGGAGGAGACCCAA
TCCTTTACCAACACTTATTCTGATTTTTTGGTCACCCTGAAGTTTACATTCTTATTCTACCAGGATTTGGTATTATTTCT
CACATTATCTCATTCCATAGAGGAAAAAAAAATCCTTTCGGAAATTTAGGTATAATCTATGCTATATTAACTATTGGAAT
TTTAGGATTCATTGTATGAGCTCATCACATATTCACTATTGGTCTTGATATTGATACT
KO-J10
AGGAACAGGATGAACAGTTTATCCCCCTCTTTCAAGAAATATTTCTCATAGAGGAGCTTCAGTTGACTTAAGAATCTTCA
GATTACATTTAGCAGGATTCTCTTCAATCCTAGGAGCTATCAACTTCATTACAACTATTTTAAATATACGACCAAAAAAC
TTAAATCTAGAATTAATCCCTTTATTTGTATGATCTGTTCTAATCACAGCAATTTTATTATTACTTAGATTACCAGTCTT
AGCAGCAGCTATTACCATACTTCTAACAGATCGTAATCTCAATACCTCATTTTTCGACCCTGCAGGTGGAGGAGACCCAA
TCCTTTACCAACACTTATTCTGATTTTTTGGTCACCCTGAAGTTTACATTCTTATTCTACCAGGATTTGGTATTATTTCT
CACATTATCTCATTCCATAGAGGAAAAAAAAATCCTTTCGGAAATTTAGGTATAATCTATGCTATATTAACTATTGGAAT
TTTAGGATTCATTGTATGAGCTCATCACATATTCACTATTGGTCTTGATATTGATACT
KO-C1
AGGAACAGGATGAACAGTTTATCCCCCTCTTTCAAGAAATATTTCTCATAGAGGAGCTTCAGTTGACTTAAGAATCTTCA
GATTACATTTAGCAGGATTCTCTTCAATCCTAGGAGCTATCAACTTCATTACAACTATTTTAAATATACGACCAAAAAAC
TTAAATCTAGAATTAATCCCTTTATTTGTATGATCTGTTCTAATCACAGCAATTTTATTATTACTTAGATTACCAGTCTT
AGCAGCAGCTATTACCATACTTCTAACAGATCGTAATCTCAATACCTCATTTTTCGACCCTGCAGGTGGAGGAGACCCAA
TCCTTTACCAACACTTATTCTGATTTTTTGGTCACCCTGAAGTTTACATTCTTATTCTACCAGGATTTGGTATTATTTCT
CACATTATCTCATTCCATAGAGGAAAAAAAAATCCTTTCGGAAATTTAGGTATAATCTATGCTATATTAACTATTGGAAT
TTTAGGATTCATTGTATGAGCTCATCACATATTCACTATTGGTCTTGATATTGATACT
KO-C2
AGGAACAGGATGAACAGTTTATCCCCCTCTTTCAAGAAATATTTCTCATAGAGGAGCTTCAGTTGACTTAAGAATCTTCA
GATTACATTTAGCAGGATTCTCTTCAATCCTAGGAGCTATCAACTTCATTACAACTATTTTAAATATACGACCAAAAAAC
TTAAATCTAGAATTAATCCCTTTATTTGTATGATCTGTTCTAATCACAGCAATTTTATTATTACTTAGATTACCAGTCTT
AGCAGCAGCTATTACCATACTTCTAACAGATCGTAATCTCAATACCTCATTTTTCGACCCTGCAGGTGGAGGAGACCCAA
TCCTTTACCAACACTTATTCTGATTTTTTGGTCACCCTGAAGTTTACATTCTTATTCTACCAGGATTTGGTATTATTTCT
CACATTATCTCATTCCATAGAGGAAAAAAAAATCCTTTCGGAAATTTAGGTATAATCTATGCTATATTAACTATTGGAAT
TTTAGGATTCATTGTATGAGCTCATCACATATTCACTATTGGTCTTGATATTGATACT
KO-C3
AGGAACAGGATGAACAGTTTATCCCCCTCTTTCAAGAAATATTTCTCATAGAGGAGCTTCAGTTGACTTAAGAATCTTCA
GATTACATTTAGCAGGATTCTCTTCAATCCTAGGAGCTATCAACTTCATTACAACTATTTTAAATATACGACCAAAAAAC
TTAAATCTAGAATTAATCCCTTTATTTGTATGATCTGTTCTAATCACAGCAATTTTATTATTACTTAGATTACCAGTCTT
AGCAGCAGCTATTACCATACTTCTAACAGATCGTAATCTCAATACCTCATTTTTCGACCCTGCAGGTGGAGGAGACCCAA
TCCTTTACCAACACTTATTCTGATTTTTTGGTCACCCTGAAGTTTACATTCTTATTCTACCAGGATTTGGTATTATTTCT
CACATTATCTCATTCCATAGAGGAAAAAAAAATCCTTTCGGAAATTTAGGTATAATCTATGCTATATTAACTATTGGAAT
TTTAGGATTCATTGTATGAGCTCATCACATATTCACTATTGGTCTTGATATTGATACT
KO-C4
AGGAACAGGATGAACAGTTTATCCCCCTCTTTCAAGAAATATTTCTCATAGAGGAGCTTCAGTTGACTTAAGAATCTTCA
GATTACATTTAGCAGGATTCTCTTCAATCCTAGGAGCTATCAACTTCATTACAACTATTTTAAATATACGACCAAAAAAC
TTAAATCTAGAATTAATCCCTTTATTTGTATGATCTGTTCTAATCACAGCAATTTTATTATTACTTAGATTACCAGTCTT
AGCAGCAGCTATTACCATACTTCTAACAGATCGTAATCTCAATACCTCATTTTTCGACCCTGCAGGTGGAGGAGACCCAA
TCCTTTACCAACACTTATTCTGATTTTTTGGTCACCCTGAAGTTTACATTCTTATTCTACCAGGATTTGGTATTATTTCT
CACATTATCTCATTCCATAGAGGAAAAAAAAATCCTTTCGGAAATTTAGGTATAATCTATGCTATATTAACTATTGGAAT
TTTAGGATTCATTGTATGAGCTCATCACATATTCACTATTGGTCTTGATATTGATACT
KO-C5
AGGAACAGGATGAACAGTTTATCCCCCTCTTTCAAGAAATATTTCTCATAGAGGAGCTTCAGTTGACTTAAGAATCTTCA
GATTACATTTAGCAGGATTCTCTTCAATCCTAGGAGCTATCAACTTCATTACAACTATTTTAAATATACGACCAAAAAAC
TTAAATCTAGAATTAATCCCTTTATTTGTATGATCTGTTCTAATCACAGCAATTTTATTATTACTTAGATTACCAGTCTT
AGCAGCAGCTATTACCATACTTCTAACAGATCGTAATCTCAATACCTCATTTTTCGACCCTGCAGGTGGAGGAGACCCAA
TCCTTTACCAACACTTATTCTGATTTTTTGGTCACCCTGAAGTTTACATTCTTATTCTACCAGGATTTGGTATTATTTCT
CACATTATCTCATTCCATAGAGGAAAAAAAAATCCTTTCGGAAATTTAGGTATAATCTATGCTATATTAACTATTGGAAT
TTTAGGATTCATTGTATGAGCTCATCACATATTCACTATTGGTCTTGATATTGATACT
KO-C6
AGGAACAGGATGAACAGTTTATCCCCCTCTTTCAAGAAATATTTCTCATAGAGGAGCTTCAGTTGACTTAAGAATCTTCA
GATTACATTTAGCAGGATTCTCTTCAATCCTAGGAGCTATCAACTTCATTACAACTATTTTAAATATACGACCAAAAAAC
TTAAATCTAGAATTAATCCCTTTATTTGTATGATCTGTTCTAATCACAGCAATTTTATTATTACTTAGATTACCAGTCTT
AGCAGCAGCTATTACCATACTTCTAACAGATCGTAATCTCAATACCTCATTTTTCGACCCTGCAGGTGGAGGAGACCCAA
TCCTTTACCAACACTTATTCTGATTTTTTGGTCACCCTGAAGTTTACATTCTTATTCTACCAGGATTTGGTATTATTTCT
CACATTATCTCATTCCATAGAGGAAAAAAAAATCCTTTCGGAAATTTAGGTATAATCTATGCTATATTAACTATTGGAAT
TTTAGGATTCATTGTATGAGCTCATCACATATTCACTATTGGTCTTGATATTGATACT
KO-C7
AGGAACAGGATGAACAGTTTATCCCCCTCTTTCAAGAAATATTTCTCATAGAGGAGCTTCAGTTGACTTAAGAATCTTCA
GATTACATTTAGCAGGATTCTCTTCAATCCTAGGAGCTATCAACTTCATTACAACTATTTTAAATATACGACCAAAAAAC
TTAAATCTAGAATTAATCCCTTTATTTGTATGATCTGTTCTAATCACAGCAATTTTATTATTACTTAGATTACCAGTCTT
AGCAGCAGCTATTACCATACTTCTAACAGATCGTAATCTCAATACCTCATTTTTCGACCCTGCAGGTGGAGGAGACCCAA
TCCTTTACCAACACTTATTCTGATTTTTTGGTCACCCTGAAGTTTACATTCTTATTCTACCAGGATTTGGTATTATTTCT
CACATTATCTCATTCCATAGAGGAAAAAAAAATCCTTTCGGAAATTTAGGTATAATCTATGCTATATTAACTATTGGAAT
TTTAGGATTCATTGTATGAGCTCATCACATATTCACTATTGGTCTTGATATTGATACT
KO-C8
AGGAACAGGATGAACAGTTTATCCCCCTCTTTCAAGAAATATTTCTCATAGAGGAGCTTCAGTTGACTTAAGAATCTTCA
GATTACATTTAGCAGGATTCTCTTCAATCCTAGGAGCTATCAACTTCATTACAACTATTTTAAATATACGACCAAAAAAC
TTAAATCTAGAATTAATCCCTTTATTTGTATGATCTGTTCTAATCACAGCAATTTTATTATTACTTAGATTACCAGTCTT
AGCAGCAGCTATTACCATACTTCTAACAGATCGTAATCTCAATACCTCATTTTTCGACCCTGCAGGTGGAGGAGACCCAA
TCCTTTACCAACACTTATTCTGATTTTTTGGTCACCCTGAAGTTTACATTCTTATTCTACCAGGATTTGGTATTATTTCT
CACATTATCTCATTCCATAGAGGAAAAAAAAATCCTTTCGGAAATTTAGGTATAATCTATGCTATATTAACTATTGGAAT
TTTAGGATTCATTGTATGAGCTCATCACATATTCACTATTGGTCTTGATATTGATACT
KO-C9
AGGAACAGGATGAACAGTTTATCCCCCTCTTTCAAGAAATATTTCTCATAGAGGAGCTTCAGTTGACTTAAGAATCTTCA
GATTACATTTAGCAGGATTCTCTTCAATCCTAGGAGCTATCAACTTCATTACAACTATTTTAAATATACGACCAAAAAAC
TTAAATCTAGAATTAATCCCTTTATTTGTATGATCTGTTCTAATCACAGCAATTTTATTATTACTTAGATTACCAGTCTT
AGCAGCAGCTATTACCATACTTCTAACAGATCGTAATCTCAATACCTCATTTTTCGACCCTGCAGGTGGAGGAGACCCAA
TCCTTTACCAACACTTATTCTGATTTTTTGGTCACCCTGAAGTTTACATTCTTATTCTACCAGGATTTGGTATTATTTCT
CACATTATCTCATTCCATAGAGGAAAAAAAAATCCTTTCGGAAATTTAGGTATAATCTATGCTATATTAACTATTGGAAT
TTTAGGATTCATTGTATGAGCTCATCACATATTCACTATTGGTCTTGATATTGATACT
KO-C10
AGGAACAGGATGAACAGTTTATCCCCCTCTTTCAAGAAATATTTCTCATAGAGGAGCTTCAGTTGACTTAAGAATCTTCA
GATTACATTTAGCAGGATTCTCTTCAATCCTAGGAGCTATCAACTTCATTACAACTATTTTAAATATACGACCAAAAAAC
TTAAATCTAGAATTAATCCCTTTATTTGTATGATCTGTTCTAATCACAGCAATTTTATTATTACTTAGATTACCAGTCTT
AGCAGCAGCTATTACCATACTTCTAACAGATCGTAATCTCAATACCTCATTTTTCGACCCTGCAGGTGGAGGAGACCCAA
TCCTTTACCAACACTTATTCTGATTTTTTGGTCACCCTGAAGTTTACATTCTTATTCTACCAGGATTTGGTATTATTTCT
CACATTATCTCATTCCATAGAGGAAAAAAAAATCCTTTCGGAAATTTAGGTATAATCTATGCTATATTAACTATTGGAAT
TTTAGGATTCATTGTATGAGCTCATCACATATTCACTATTGGTCTTGATATTGATACT
UZ-TC01
AGGAACAGGATGAACAATTTATCCCCCTCTTTCAAGAAATATTTCTCATAGAGGAGCTTCAGTTGACTTAAGAATCTTCA
GATTACATTTAGCAGGATTCTCTTCAATCCTAGGAGCTATCAACTTCATTACAACTATTTTAAATATACGACCAAAAAAC
TTAAATCTAGAATTAATCCCTTTATTTGTATGATCTGTTCTAATCACAGCAATTTTATTATTACTTAGATTACCAGTCTT
AGCAGCAGCTATTACCATACTTCTAACAGATCGTAATCTCAATACCTCATTTTTCGACCCTGCAGGTGGAGGAGACCCAA
TCCTTTACCAACACTTATTCTGATTTTTTGGTCACCCTGAAGTTTACATTCTTATTCTACCAGGATTTGGTATTATTTCT
CACATTATCTCATTCCATAGAGGAAAAAAAAATCCTTTCGGAAATTTAGGTATAATCTATGCTATATTAACTATTGGAAT
TTTAGGATTCATTGTATGAGCTCATCACATATTCACTATTGGTCTTGATATTGATACT
UZ-TC02
AGGAACAGGATGAACAATTTATCCCCCTCTTTCAAGAAATATTTCTCATAGAGGAGCTTCAGTTGACTTAAGAATCTTCA
GATTACATTTAGCAGGATTCTCTTCAATCCTAGGAGCTATCAACTTCATTACAACTATTTTAAATATACGACCAAAAAAC
TTAAATCTAGAATTAATCCCTTTATTTGTATGATCTGTTCTAATCACAGCAATTTTATTATTACTTAGATTACCAGTCTT
AGCAGCAGCTATTACCATACTTCTAACAGATCGTAATCTCAATACCTCATTTTTCGACCCTGCAGGTGGAGGAGACCCAA
TCCTTTACCAACACTTATTCTGATTTTTTGGTCACCCTGAAGTTTACATTCTTATTCTACCAGGATTTGGTATTATTTCT
CACATTATCTCATTCCATAGAGGAAAAAAAAATCCTTTCGGAAATTTAGGTATAATCTATGCTATATTAACTATTGGAAT
TTTAGGATTCATTGTATGAGCTCATCACATATTCACTATTGGTCTTGATATTGATACT
UZ-TC03
AGGAACAGGATGAACAATTTATCCCCCTCTTTCAAGAAATATTTCTCATAGAGGAGCTTCAGTTGACTTAAGAATCTTCA
GATTACATTTAGCAGGATTCTCTTCAATCCTAGGAGCTATCAACTTCATTACAACTATTTTAAATATACGACCAAAAAAC
TTAAATCTAGAATTAATCCCTTTATTTGTATGATCTGTTCTAATCACAGCAATTTTATTATTACTTAGATTACCAGTCTT
AGCAGCAGCTATTACCATACTTCTAACAGATCGTAATCTCAATACCTCATTTTTCGACCCTGCAGGTGGAGGAGACCCAA
TCCTTTACCAACACTTATTCTGATTTTTTGGTCACCCTGAAGTTTACATTCTTATTCTACCAGGATTTGGTATTATTTCT
CACATTATCTCATTCCATAGAGGAAAAAAAAATCCTTTCGGAAATTTAGGTATAATCTATGCTATATTAACTATTGGAAT
TTTAGGATTCATTGTATGAGCTCATCACATATTCACTATTGGTCTTGATATTGATACT
    \end{Verbatim}

    \subsubsection{Generate 20 random protein sequences of each sequences of
input fiel at least 30\% identical from input file
input.fasta}\label{generate-20-random-protein-sequences-of-each-sequences-of-input-fiel-at-least-30-identical-from-input-file-input.fasta}

    \begin{tcolorbox}[breakable, size=fbox, boxrule=1pt, pad at break*=1mm,colback=cellbackground, colframe=cellborder]
\prompt{In}{incolor}{19}{\boxspacing}
\begin{Verbatim}[commandchars=\\\{\}]
\PY{k}{def} \PY{n+nf}{mutate\PYZus{}sequence}\PY{p}{(}\PY{n}{original\PYZus{}seq}\PY{p}{,} \PY{n}{identity\PYZus{}percent}\PY{o}{=}\PY{l+m+mi}{30}\PY{p}{)}\PY{p}{:}
    \PY{n}{seq\PYZus{}length} \PY{o}{=} \PY{n+nb}{len}\PY{p}{(}\PY{n}{original\PYZus{}seq}\PY{p}{)}
    \PY{n}{num\PYZus{}mutations} \PY{o}{=} \PY{n+nb}{int}\PY{p}{(}\PY{p}{(}\PY{l+m+mi}{100} \PY{o}{\PYZhy{}} \PY{n}{identity\PYZus{}percent}\PY{p}{)} \PY{o}{/} \PY{l+m+mi}{100} \PY{o}{*} \PY{n}{seq\PYZus{}length}\PY{p}{)}
    \PY{n}{original\PYZus{}seq} \PY{o}{=} \PY{n+nb}{list}\PY{p}{(}\PY{n}{original\PYZus{}seq}\PY{p}{)}
    \PY{n}{amino\PYZus{}acids} \PY{o}{=} \PY{p}{[}\PY{l+s+s1}{\PYZsq{}}\PY{l+s+s1}{A}\PY{l+s+s1}{\PYZsq{}}\PY{p}{,} \PY{l+s+s1}{\PYZsq{}}\PY{l+s+s1}{R}\PY{l+s+s1}{\PYZsq{}}\PY{p}{,} \PY{l+s+s1}{\PYZsq{}}\PY{l+s+s1}{N}\PY{l+s+s1}{\PYZsq{}}\PY{p}{,} \PY{l+s+s1}{\PYZsq{}}\PY{l+s+s1}{D}\PY{l+s+s1}{\PYZsq{}}\PY{p}{,} \PY{l+s+s1}{\PYZsq{}}\PY{l+s+s1}{C}\PY{l+s+s1}{\PYZsq{}}\PY{p}{,} \PY{l+s+s1}{\PYZsq{}}\PY{l+s+s1}{Q}\PY{l+s+s1}{\PYZsq{}}\PY{p}{,} \PY{l+s+s1}{\PYZsq{}}\PY{l+s+s1}{E}\PY{l+s+s1}{\PYZsq{}}\PY{p}{,} \PY{l+s+s1}{\PYZsq{}}\PY{l+s+s1}{G}\PY{l+s+s1}{\PYZsq{}}\PY{p}{,} \PY{l+s+s1}{\PYZsq{}}\PY{l+s+s1}{H}\PY{l+s+s1}{\PYZsq{}}\PY{p}{,} \PY{l+s+s1}{\PYZsq{}}\PY{l+s+s1}{I}\PY{l+s+s1}{\PYZsq{}}\PY{p}{,} \PY{l+s+s1}{\PYZsq{}}\PY{l+s+s1}{L}\PY{l+s+s1}{\PYZsq{}}\PY{p}{,} \PY{l+s+s1}{\PYZsq{}}\PY{l+s+s1}{K}\PY{l+s+s1}{\PYZsq{}}\PY{p}{,} \PY{l+s+s1}{\PYZsq{}}\PY{l+s+s1}{M}\PY{l+s+s1}{\PYZsq{}}\PY{p}{,} \PY{l+s+s1}{\PYZsq{}}\PY{l+s+s1}{F}\PY{l+s+s1}{\PYZsq{}}\PY{p}{,} \PY{l+s+s1}{\PYZsq{}}\PY{l+s+s1}{P}\PY{l+s+s1}{\PYZsq{}}\PY{p}{,} \PY{l+s+s1}{\PYZsq{}}\PY{l+s+s1}{S}\PY{l+s+s1}{\PYZsq{}}\PY{p}{,} \PY{l+s+s1}{\PYZsq{}}\PY{l+s+s1}{T}\PY{l+s+s1}{\PYZsq{}}\PY{p}{,} \PY{l+s+s1}{\PYZsq{}}\PY{l+s+s1}{W}\PY{l+s+s1}{\PYZsq{}}\PY{p}{,} \PY{l+s+s1}{\PYZsq{}}\PY{l+s+s1}{Y}\PY{l+s+s1}{\PYZsq{}}\PY{p}{,} \PY{l+s+s1}{\PYZsq{}}\PY{l+s+s1}{V}\PY{l+s+s1}{\PYZsq{}}\PY{p}{]}
    
    \PY{k}{for} \PY{n}{\PYZus{}} \PY{o+ow}{in} \PY{n+nb}{range}\PY{p}{(}\PY{n}{num\PYZus{}mutations}\PY{p}{)}\PY{p}{:}
        \PY{n}{index\PYZus{}to\PYZus{}mutate} \PY{o}{=} \PY{n}{random}\PY{o}{.}\PY{n}{randint}\PY{p}{(}\PY{l+m+mi}{0}\PY{p}{,} \PY{n}{seq\PYZus{}length} \PY{o}{\PYZhy{}} \PY{l+m+mi}{1}\PY{p}{)}
        \PY{n}{new\PYZus{}amino\PYZus{}acid} \PY{o}{=} \PY{n}{random}\PY{o}{.}\PY{n}{choice}\PY{p}{(}\PY{n}{amino\PYZus{}acids}\PY{p}{)}  \PY{c+c1}{\PYZsh{} Selecting a new amino acid}
        \PY{k}{while} \PY{n}{original\PYZus{}seq}\PY{p}{[}\PY{n}{index\PYZus{}to\PYZus{}mutate}\PY{p}{]} \PY{o}{==} \PY{n}{new\PYZus{}amino\PYZus{}acid}\PY{p}{:}
            \PY{n}{new\PYZus{}amino\PYZus{}acid} \PY{o}{=} \PY{n}{random}\PY{o}{.}\PY{n}{choice}\PY{p}{(}\PY{n}{amino\PYZus{}acids}\PY{p}{)}
        \PY{n}{original\PYZus{}seq}\PY{p}{[}\PY{n}{index\PYZus{}to\PYZus{}mutate}\PY{p}{]} \PY{o}{=} \PY{n}{new\PYZus{}amino\PYZus{}acid}
    
    \PY{k}{return} \PY{l+s+s1}{\PYZsq{}}\PY{l+s+s1}{\PYZsq{}}\PY{o}{.}\PY{n}{join}\PY{p}{(}\PY{n}{original\PYZus{}seq}\PY{p}{)}


\PY{n}{fasta\PYZus{}file} \PY{o}{=} \PY{l+s+s1}{\PYZsq{}}\PY{l+s+s1}{input.fasta}\PY{l+s+s1}{\PYZsq{}}
\PY{n}{sequences} \PY{o}{=} \PY{n+nb}{list}\PY{p}{(}\PY{n}{SeqIO}\PY{o}{.}\PY{n}{parse}\PY{p}{(}\PY{n}{fasta\PYZus{}file}\PY{p}{,} \PY{l+s+s2}{\PYZdq{}}\PY{l+s+s2}{fasta}\PY{l+s+s2}{\PYZdq{}}\PY{p}{)}\PY{p}{)}
\PY{n}{new\PYZus{}records} \PY{o}{=} \PY{p}{[}\PY{p}{]}

\PY{k}{for} \PY{n}{seq} \PY{o+ow}{in} \PY{n}{sequences}\PY{p}{:}
    \PY{k}{for} \PY{n}{i} \PY{o+ow}{in} \PY{n+nb}{range}\PY{p}{(}\PY{l+m+mi}{20}\PY{p}{)}\PY{p}{:}  \PY{c+c1}{\PYZsh{} Generate 20 mutated sequences per original}
        \PY{n}{new\PYZus{}seq} \PY{o}{=} \PY{n}{mutate\PYZus{}sequence}\PY{p}{(}\PY{n}{seq}\PY{o}{.}\PY{n}{seq}\PY{p}{,} \PY{n}{identity\PYZus{}percent}\PY{o}{=}\PY{l+m+mi}{70}\PY{p}{)}  \PY{c+c1}{\PYZsh{} Maintain 70\PYZpc{} identity}
        \PY{n}{new\PYZus{}record} \PY{o}{=} \PY{n}{SeqRecord}\PY{p}{(}\PY{n}{Seq}\PY{p}{(}\PY{n}{new\PYZus{}seq}\PY{p}{)}\PY{p}{,} \PY{n+nb}{id}\PY{o}{=}\PY{l+s+sa}{f}\PY{l+s+s2}{\PYZdq{}}\PY{l+s+si}{\PYZob{}}\PY{n}{seq}\PY{o}{.}\PY{n}{id}\PY{l+s+si}{\PYZcb{}}\PY{l+s+s2}{\PYZus{}mutated\PYZus{}}\PY{l+s+si}{\PYZob{}}\PY{n}{i}\PY{o}{+}\PY{l+m+mi}{1}\PY{l+s+si}{\PYZcb{}}\PY{l+s+s2}{\PYZdq{}}\PY{p}{,} \PY{n}{description}\PY{o}{=}\PY{l+s+s2}{\PYZdq{}}\PY{l+s+s2}{\PYZdq{}}\PY{p}{)}
        \PY{n}{new\PYZus{}records}\PY{o}{.}\PY{n}{append}\PY{p}{(}\PY{n}{new\PYZus{}record}\PY{p}{)}

\PY{n}{output\PYZus{}file} \PY{o}{=} \PY{l+s+s1}{\PYZsq{}}\PY{l+s+s1}{output.fasta}\PY{l+s+s1}{\PYZsq{}}
\PY{k}{with} \PY{n+nb}{open}\PY{p}{(}\PY{n}{output\PYZus{}file}\PY{p}{,} \PY{l+s+s1}{\PYZsq{}}\PY{l+s+s1}{w}\PY{l+s+s1}{\PYZsq{}}\PY{p}{)} \PY{k}{as} \PY{n}{f}\PY{p}{:}
    \PY{n}{SeqIO}\PY{o}{.}\PY{n}{write}\PY{p}{(}\PY{n}{new\PYZus{}records}\PY{p}{,} \PY{n}{f}\PY{p}{,} \PY{l+s+s1}{\PYZsq{}}\PY{l+s+s1}{fasta}\PY{l+s+s1}{\PYZsq{}}\PY{p}{)}
\end{Verbatim}
\end{tcolorbox}

    \section{check for Alignment}\label{check-for-alignment}

    \begin{tcolorbox}[breakable, size=fbox, boxrule=1pt, pad at break*=1mm,colback=cellbackground, colframe=cellborder]
\prompt{In}{incolor}{20}{\boxspacing}
\begin{Verbatim}[commandchars=\\\{\}]
\PY{c+ch}{\PYZsh{}!/usr/bin/python3}
\PY{k}{def} \PY{n+nf}{check\PYZus{}alignment}\PY{p}{(}\PY{n}{file}\PY{p}{)}\PY{p}{:}
    \PY{n}{sequences} \PY{o}{=} \PY{p}{[}\PY{n+nb}{str}\PY{p}{(}\PY{n}{record}\PY{o}{.}\PY{n}{seq}\PY{p}{)} \PY{k}{for} \PY{n}{record} \PY{o+ow}{in} \PY{n}{SeqIO}\PY{o}{.}\PY{n}{parse}\PY{p}{(}\PY{n}{file}\PY{p}{,} \PY{l+s+s2}{\PYZdq{}}\PY{l+s+s2}{fasta}\PY{l+s+s2}{\PYZdq{}}\PY{p}{)}\PY{p}{]}
    \PY{n}{length} \PY{o}{=} \PY{n+nb}{len}\PY{p}{(}\PY{n}{sequences}\PY{p}{[}\PY{l+m+mi}{0}\PY{p}{]}\PY{p}{)}
    \PY{k}{for} \PY{n}{seq} \PY{o+ow}{in} \PY{n}{sequences}\PY{p}{:}
        \PY{k}{if} \PY{n+nb}{len}\PY{p}{(}\PY{n}{seq}\PY{p}{)} \PY{o}{!=} \PY{n}{length}\PY{p}{:}
            \PY{k}{return} \PY{k+kc}{False}
    \PY{k}{return} \PY{k+kc}{True}

\PY{n+nb}{print}\PY{p}{(}\PY{n}{check\PYZus{}alignment}\PY{p}{(}\PY{l+s+s2}{\PYZdq{}}\PY{l+s+s2}{output.fasta}\PY{l+s+s2}{\PYZdq{}}\PY{p}{)}\PY{p}{)}
\end{Verbatim}
\end{tcolorbox}

    \begin{Verbatim}[commandchars=\\\{\}]
True
    \end{Verbatim}

    \section{check for semilarity}\label{check-for-semilarity}

\begin{itemize}
\tightlist
\item
  compares each original sequence in the input file with its
  corresponding mutated versions in the output file
\end{itemize}

    \begin{tcolorbox}[breakable, size=fbox, boxrule=1pt, pad at break*=1mm,colback=cellbackground, colframe=cellborder]
\prompt{In}{incolor}{21}{\boxspacing}
\begin{Verbatim}[commandchars=\\\{\}]
\PY{k+kn}{from} \PY{n+nn}{Bio} \PY{k+kn}{import} \PY{n}{SeqIO}
\PY{k+kn}{from} \PY{n+nn}{Bio}\PY{n+nn}{.}\PY{n+nn}{pairwise2} \PY{k+kn}{import} \PY{n}{align}

\PY{c+c1}{\PYZsh{} Read sequences from the input and output FASTA files}
\PY{n}{input\PYZus{}fasta\PYZus{}file} \PY{o}{=} \PY{l+s+s1}{\PYZsq{}}\PY{l+s+s1}{input.fasta}\PY{l+s+s1}{\PYZsq{}}
\PY{n}{output\PYZus{}fasta\PYZus{}file} \PY{o}{=} \PY{l+s+s1}{\PYZsq{}}\PY{l+s+s1}{output.fasta}\PY{l+s+s1}{\PYZsq{}}
\PY{n}{input\PYZus{}sequences} \PY{o}{=} \PY{n+nb}{list}\PY{p}{(}\PY{n}{SeqIO}\PY{o}{.}\PY{n}{parse}\PY{p}{(}\PY{n}{input\PYZus{}fasta\PYZus{}file}\PY{p}{,} \PY{l+s+s2}{\PYZdq{}}\PY{l+s+s2}{fasta}\PY{l+s+s2}{\PYZdq{}}\PY{p}{)}\PY{p}{)}
\PY{n}{output\PYZus{}sequences} \PY{o}{=} \PY{n+nb}{list}\PY{p}{(}\PY{n}{SeqIO}\PY{o}{.}\PY{n}{parse}\PY{p}{(}\PY{n}{output\PYZus{}fasta\PYZus{}file}\PY{p}{,} \PY{l+s+s2}{\PYZdq{}}\PY{l+s+s2}{fasta}\PY{l+s+s2}{\PYZdq{}}\PY{p}{)}\PY{p}{)}

\PY{c+c1}{\PYZsh{} Function to calculate similarity between two sequences}
\PY{k}{def} \PY{n+nf}{calculate\PYZus{}similarity}\PY{p}{(}\PY{n}{seq1}\PY{p}{,} \PY{n}{seq2}\PY{p}{)}\PY{p}{:}
    \PY{n}{alignments} \PY{o}{=} \PY{n}{align}\PY{o}{.}\PY{n}{globalxx}\PY{p}{(}\PY{n}{seq1}\PY{o}{.}\PY{n}{seq}\PY{p}{,} \PY{n}{seq2}\PY{o}{.}\PY{n}{seq}\PY{p}{)}
    \PY{c+c1}{\PYZsh{} Get the highest scoring alignment}
    \PY{n}{best\PYZus{}alignment} \PY{o}{=} \PY{n+nb}{max}\PY{p}{(}\PY{n}{alignments}\PY{p}{,} \PY{n}{key}\PY{o}{=}\PY{k}{lambda} \PY{n}{x}\PY{p}{:} \PY{n}{x}\PY{o}{.}\PY{n}{score}\PY{p}{)}
    \PY{c+c1}{\PYZsh{} Calculate similarity percentage}
    \PY{n}{similarity} \PY{o}{=} \PY{n}{best\PYZus{}alignment}\PY{o}{.}\PY{n}{score} \PY{o}{/} \PY{n+nb}{min}\PY{p}{(}\PY{n+nb}{len}\PY{p}{(}\PY{n}{seq1}\PY{o}{.}\PY{n}{seq}\PY{p}{)}\PY{p}{,} \PY{n+nb}{len}\PY{p}{(}\PY{n}{seq2}\PY{o}{.}\PY{n}{seq}\PY{p}{)}\PY{p}{)} \PY{o}{*} \PY{l+m+mi}{100}
    \PY{k}{return} \PY{n}{similarity}

\PY{c+c1}{\PYZsh{} Mapping original sequences to their mutations}
\PY{n}{mutations\PYZus{}map} \PY{o}{=} \PY{p}{\PYZob{}}\PY{p}{\PYZcb{}}
\PY{k}{for} \PY{n}{seq} \PY{o+ow}{in} \PY{n}{output\PYZus{}sequences}\PY{p}{:}
    \PY{n}{original\PYZus{}id} \PY{o}{=} \PY{n}{seq}\PY{o}{.}\PY{n}{id}\PY{o}{.}\PY{n}{split}\PY{p}{(}\PY{l+s+s1}{\PYZsq{}}\PY{l+s+s1}{\PYZus{}mutated}\PY{l+s+s1}{\PYZsq{}}\PY{p}{)}\PY{p}{[}\PY{l+m+mi}{0}\PY{p}{]}  \PY{c+c1}{\PYZsh{} Assumes IDs are of the form \PYZdq{}MINE01\PYZus{}mutated\PYZus{}1\PYZdq{}}
    \PY{k}{if} \PY{n}{original\PYZus{}id} \PY{o+ow}{not} \PY{o+ow}{in} \PY{n}{mutations\PYZus{}map}\PY{p}{:}
        \PY{n}{mutations\PYZus{}map}\PY{p}{[}\PY{n}{original\PYZus{}id}\PY{p}{]} \PY{o}{=} \PY{p}{[}\PY{p}{]}
    \PY{n}{mutations\PYZus{}map}\PY{p}{[}\PY{n}{original\PYZus{}id}\PY{p}{]}\PY{o}{.}\PY{n}{append}\PY{p}{(}\PY{n}{seq}\PY{p}{)}

\PY{c+c1}{\PYZsh{} Compare each original sequence with its mutated versions}
\PY{k}{for} \PY{n}{original\PYZus{}seq} \PY{o+ow}{in} \PY{n}{input\PYZus{}sequences}\PY{p}{:}
    \PY{n}{original\PYZus{}id} \PY{o}{=} \PY{n}{original\PYZus{}seq}\PY{o}{.}\PY{n}{id}
    \PY{k}{if} \PY{n}{original\PYZus{}id} \PY{o+ow}{in} \PY{n}{mutations\PYZus{}map}\PY{p}{:}
        \PY{k}{for} \PY{n}{mutated\PYZus{}seq} \PY{o+ow}{in} \PY{n}{mutations\PYZus{}map}\PY{p}{[}\PY{n}{original\PYZus{}id}\PY{p}{]}\PY{p}{:}
            \PY{n}{similarity} \PY{o}{=} \PY{n}{calculate\PYZus{}similarity}\PY{p}{(}\PY{n}{original\PYZus{}seq}\PY{p}{,} \PY{n}{mutated\PYZus{}seq}\PY{p}{)}
            \PY{n+nb}{print}\PY{p}{(}\PY{l+s+sa}{f}\PY{l+s+s2}{\PYZdq{}}\PY{l+s+s2}{Similarity between }\PY{l+s+si}{\PYZob{}}\PY{n}{original\PYZus{}id}\PY{l+s+si}{\PYZcb{}}\PY{l+s+s2}{ and }\PY{l+s+si}{\PYZob{}}\PY{n}{mutated\PYZus{}seq}\PY{o}{.}\PY{n}{id}\PY{l+s+si}{\PYZcb{}}\PY{l+s+s2}{: }\PY{l+s+si}{\PYZob{}}\PY{n}{similarity}\PY{l+s+si}{:}\PY{l+s+s2}{.2f}\PY{l+s+si}{\PYZcb{}}\PY{l+s+s2}{\PYZpc{}}\PY{l+s+s2}{\PYZdq{}}\PY{p}{)}
\end{Verbatim}
\end{tcolorbox}

    \begin{Verbatim}[commandchars=\\\{\}]
Similarity between MINE01 and MINE01\_mutated\_1: 74.72\%
Similarity between MINE01 and MINE01\_mutated\_2: 75.46\%
Similarity between MINE01 and MINE01\_mutated\_3: 74.72\%
Similarity between MINE01 and MINE01\_mutated\_4: 75.84\%
Similarity between MINE01 and MINE01\_mutated\_5: 76.02\%
Similarity between MINE01 and MINE01\_mutated\_6: 75.09\%
Similarity between MINE01 and MINE01\_mutated\_7: 74.72\%
Similarity between MINE01 and MINE01\_mutated\_8: 76.02\%
Similarity between MINE01 and MINE01\_mutated\_9: 74.72\%
Similarity between MINE01 and MINE01\_mutated\_10: 75.28\%
Similarity between MINE01 and MINE01\_mutated\_11: 75.28\%
Similarity between MINE01 and MINE01\_mutated\_12: 76.58\%
Similarity between MINE01 and MINE01\_mutated\_13: 76.39\%
Similarity between MINE01 and MINE01\_mutated\_14: 74.72\%
Similarity between MINE01 and MINE01\_mutated\_15: 75.09\%
Similarity between MINE01 and MINE01\_mutated\_16: 73.61\%
Similarity between MINE01 and MINE01\_mutated\_17: 76.02\%
Similarity between MINE01 and MINE01\_mutated\_18: 74.54\%
Similarity between MINE01 and MINE01\_mutated\_19: 76.21\%
Similarity between MINE01 and MINE01\_mutated\_20: 75.09\%
Similarity between MINE03 and MINE03\_mutated\_1: 73.98\%
Similarity between MINE03 and MINE03\_mutated\_2: 76.02\%
Similarity between MINE03 and MINE03\_mutated\_3: 73.61\%
Similarity between MINE03 and MINE03\_mutated\_4: 77.32\%
Similarity between MINE03 and MINE03\_mutated\_5: 73.61\%
Similarity between MINE03 and MINE03\_mutated\_6: 75.65\%
Similarity between MINE03 and MINE03\_mutated\_7: 73.98\%
Similarity between MINE03 and MINE03\_mutated\_8: 75.84\%
Similarity between MINE03 and MINE03\_mutated\_9: 75.46\%
Similarity between MINE03 and MINE03\_mutated\_10: 75.46\%
Similarity between MINE03 and MINE03\_mutated\_11: 75.46\%
Similarity between MINE03 and MINE03\_mutated\_12: 76.58\%
Similarity between MINE03 and MINE03\_mutated\_13: 73.05\%
Similarity between MINE03 and MINE03\_mutated\_14: 76.95\%
Similarity between MINE03 and MINE03\_mutated\_15: 75.09\%
Similarity between MINE03 and MINE03\_mutated\_16: 74.91\%
Similarity between MINE03 and MINE03\_mutated\_17: 76.02\%
Similarity between MINE03 and MINE03\_mutated\_18: 74.72\%
Similarity between MINE03 and MINE03\_mutated\_19: 76.02\%
Similarity between MINE03 and MINE03\_mutated\_20: 75.46\%
Similarity between MINE04 and MINE04\_mutated\_1: 75.09\%
Similarity between MINE04 and MINE04\_mutated\_2: 75.09\%
Similarity between MINE04 and MINE04\_mutated\_3: 73.23\%
Similarity between MINE04 and MINE04\_mutated\_4: 75.28\%
Similarity between MINE04 and MINE04\_mutated\_5: 74.72\%
Similarity between MINE04 and MINE04\_mutated\_6: 75.84\%
Similarity between MINE04 and MINE04\_mutated\_7: 74.91\%
Similarity between MINE04 and MINE04\_mutated\_8: 74.72\%
Similarity between MINE04 and MINE04\_mutated\_9: 75.84\%
Similarity between MINE04 and MINE04\_mutated\_10: 74.54\%
Similarity between MINE04 and MINE04\_mutated\_11: 76.02\%
Similarity between MINE04 and MINE04\_mutated\_12: 75.65\%
Similarity between MINE04 and MINE04\_mutated\_13: 73.42\%
Similarity between MINE04 and MINE04\_mutated\_14: 74.91\%
Similarity between MINE04 and MINE04\_mutated\_15: 75.09\%
Similarity between MINE04 and MINE04\_mutated\_16: 76.02\%
Similarity between MINE04 and MINE04\_mutated\_17: 73.79\%
Similarity between MINE04 and MINE04\_mutated\_18: 74.91\%
Similarity between MINE04 and MINE04\_mutated\_19: 75.46\%
Similarity between MINE04 and MINE04\_mutated\_20: 74.91\%
Similarity between MINE05 and MINE05\_mutated\_1: 74.72\%
Similarity between MINE05 and MINE05\_mutated\_2: 74.35\%
Similarity between MINE05 and MINE05\_mutated\_3: 74.35\%
Similarity between MINE05 and MINE05\_mutated\_4: 75.65\%
Similarity between MINE05 and MINE05\_mutated\_5: 74.35\%
Similarity between MINE05 and MINE05\_mutated\_6: 76.02\%
Similarity between MINE05 and MINE05\_mutated\_7: 74.35\%
Similarity between MINE05 and MINE05\_mutated\_8: 75.65\%
Similarity between MINE05 and MINE05\_mutated\_9: 75.09\%
Similarity between MINE05 and MINE05\_mutated\_10: 75.46\%
Similarity between MINE05 and MINE05\_mutated\_11: 76.02\%
Similarity between MINE05 and MINE05\_mutated\_12: 74.91\%
Similarity between MINE05 and MINE05\_mutated\_13: 75.46\%
Similarity between MINE05 and MINE05\_mutated\_14: 73.79\%
Similarity between MINE05 and MINE05\_mutated\_15: 74.91\%
Similarity between MINE05 and MINE05\_mutated\_16: 76.02\%
Similarity between MINE05 and MINE05\_mutated\_17: 75.28\%
Similarity between MINE05 and MINE05\_mutated\_18: 75.28\%
Similarity between MINE05 and MINE05\_mutated\_19: 74.72\%
Similarity between MINE05 and MINE05\_mutated\_20: 74.54\%
Similarity between MINE06 and MINE06\_mutated\_1: 74.16\%
Similarity between MINE06 and MINE06\_mutated\_2: 75.09\%
Similarity between MINE06 and MINE06\_mutated\_3: 75.46\%
Similarity between MINE06 and MINE06\_mutated\_4: 75.28\%
Similarity between MINE06 and MINE06\_mutated\_5: 75.46\%
Similarity between MINE06 and MINE06\_mutated\_6: 75.09\%
Similarity between MINE06 and MINE06\_mutated\_7: 74.16\%
Similarity between MINE06 and MINE06\_mutated\_8: 76.95\%
Similarity between MINE06 and MINE06\_mutated\_9: 74.54\%
Similarity between MINE06 and MINE06\_mutated\_10: 74.54\%
Similarity between MINE06 and MINE06\_mutated\_11: 74.16\%
Similarity between MINE06 and MINE06\_mutated\_12: 73.61\%
Similarity between MINE06 and MINE06\_mutated\_13: 75.09\%
Similarity between MINE06 and MINE06\_mutated\_14: 75.46\%
Similarity between MINE06 and MINE06\_mutated\_15: 75.65\%
Similarity between MINE06 and MINE06\_mutated\_16: 74.91\%
Similarity between MINE06 and MINE06\_mutated\_17: 73.79\%
Similarity between MINE06 and MINE06\_mutated\_18: 75.65\%
Similarity between MINE06 and MINE06\_mutated\_19: 74.35\%
Similarity between MINE06 and MINE06\_mutated\_20: 75.28\%
Similarity between KO-A1 and KO-A1\_mutated\_1: 74.16\%
Similarity between KO-A1 and KO-A1\_mutated\_2: 74.54\%
Similarity between KO-A1 and KO-A1\_mutated\_3: 74.16\%
Similarity between KO-A1 and KO-A1\_mutated\_4: 75.84\%
Similarity between KO-A1 and KO-A1\_mutated\_5: 75.46\%
Similarity between KO-A1 and KO-A1\_mutated\_6: 75.65\%
Similarity between KO-A1 and KO-A1\_mutated\_7: 75.46\%
Similarity between KO-A1 and KO-A1\_mutated\_8: 75.28\%
Similarity between KO-A1 and KO-A1\_mutated\_9: 74.91\%
Similarity between KO-A1 and KO-A1\_mutated\_10: 74.91\%
Similarity between KO-A1 and KO-A1\_mutated\_11: 74.72\%
Similarity between KO-A1 and KO-A1\_mutated\_12: 75.65\%
Similarity between KO-A1 and KO-A1\_mutated\_13: 73.79\%
Similarity between KO-A1 and KO-A1\_mutated\_14: 74.54\%
Similarity between KO-A1 and KO-A1\_mutated\_15: 73.98\%
Similarity between KO-A1 and KO-A1\_mutated\_16: 74.72\%
Similarity between KO-A1 and KO-A1\_mutated\_17: 73.98\%
Similarity between KO-A1 and KO-A1\_mutated\_18: 74.35\%
Similarity between KO-A1 and KO-A1\_mutated\_19: 75.09\%
Similarity between KO-A1 and KO-A1\_mutated\_20: 73.79\%
Similarity between KO-A2 and KO-A2\_mutated\_1: 74.72\%
Similarity between KO-A2 and KO-A2\_mutated\_2: 74.54\%
Similarity between KO-A2 and KO-A2\_mutated\_3: 74.54\%
Similarity between KO-A2 and KO-A2\_mutated\_4: 75.84\%
Similarity between KO-A2 and KO-A2\_mutated\_5: 74.72\%
Similarity between KO-A2 and KO-A2\_mutated\_6: 76.02\%
Similarity between KO-A2 and KO-A2\_mutated\_7: 75.46\%
Similarity between KO-A2 and KO-A2\_mutated\_8: 75.09\%
Similarity between KO-A2 and KO-A2\_mutated\_9: 75.28\%
Similarity between KO-A2 and KO-A2\_mutated\_10: 74.54\%
Similarity between KO-A2 and KO-A2\_mutated\_11: 73.79\%
Similarity between KO-A2 and KO-A2\_mutated\_12: 73.79\%
Similarity between KO-A2 and KO-A2\_mutated\_13: 73.79\%
Similarity between KO-A2 and KO-A2\_mutated\_14: 74.91\%
Similarity between KO-A2 and KO-A2\_mutated\_15: 74.16\%
Similarity between KO-A2 and KO-A2\_mutated\_16: 75.65\%
Similarity between KO-A2 and KO-A2\_mutated\_17: 76.02\%
Similarity between KO-A2 and KO-A2\_mutated\_18: 74.72\%
Similarity between KO-A2 and KO-A2\_mutated\_19: 76.02\%
Similarity between KO-A2 and KO-A2\_mutated\_20: 74.35\%
Similarity between KO-A3 and KO-A3\_mutated\_1: 74.91\%
Similarity between KO-A3 and KO-A3\_mutated\_2: 74.91\%
Similarity between KO-A3 and KO-A3\_mutated\_3: 74.16\%
Similarity between KO-A3 and KO-A3\_mutated\_4: 73.23\%
Similarity between KO-A3 and KO-A3\_mutated\_5: 74.54\%
Similarity between KO-A3 and KO-A3\_mutated\_6: 75.09\%
Similarity between KO-A3 and KO-A3\_mutated\_7: 75.65\%
Similarity between KO-A3 and KO-A3\_mutated\_8: 74.35\%
Similarity between KO-A3 and KO-A3\_mutated\_9: 75.65\%
Similarity between KO-A3 and KO-A3\_mutated\_10: 74.35\%
Similarity between KO-A3 and KO-A3\_mutated\_11: 73.98\%
Similarity between KO-A3 and KO-A3\_mutated\_12: 75.84\%
Similarity between KO-A3 and KO-A3\_mutated\_13: 73.79\%
Similarity between KO-A3 and KO-A3\_mutated\_14: 75.46\%
Similarity between KO-A3 and KO-A3\_mutated\_15: 74.35\%
Similarity between KO-A3 and KO-A3\_mutated\_16: 76.77\%
Similarity between KO-A3 and KO-A3\_mutated\_17: 75.28\%
Similarity between KO-A3 and KO-A3\_mutated\_18: 74.16\%
Similarity between KO-A3 and KO-A3\_mutated\_19: 75.28\%
Similarity between KO-A3 and KO-A3\_mutated\_20: 76.39\%
Similarity between KO-A4 and KO-A4\_mutated\_1: 73.98\%
Similarity between KO-A4 and KO-A4\_mutated\_2: 75.28\%
Similarity between KO-A4 and KO-A4\_mutated\_3: 76.77\%
Similarity between KO-A4 and KO-A4\_mutated\_4: 76.21\%
Similarity between KO-A4 and KO-A4\_mutated\_5: 75.46\%
Similarity between KO-A4 and KO-A4\_mutated\_6: 74.54\%
Similarity between KO-A4 and KO-A4\_mutated\_7: 75.65\%
Similarity between KO-A4 and KO-A4\_mutated\_8: 74.54\%
Similarity between KO-A4 and KO-A4\_mutated\_9: 75.65\%
Similarity between KO-A4 and KO-A4\_mutated\_10: 74.72\%
Similarity between KO-A4 and KO-A4\_mutated\_11: 75.09\%
Similarity between KO-A4 and KO-A4\_mutated\_12: 75.65\%
Similarity between KO-A4 and KO-A4\_mutated\_13: 75.84\%
Similarity between KO-A4 and KO-A4\_mutated\_14: 75.28\%
Similarity between KO-A4 and KO-A4\_mutated\_15: 74.91\%
Similarity between KO-A4 and KO-A4\_mutated\_16: 74.72\%
Similarity between KO-A4 and KO-A4\_mutated\_17: 75.46\%
Similarity between KO-A4 and KO-A4\_mutated\_18: 73.98\%
Similarity between KO-A4 and KO-A4\_mutated\_19: 75.46\%
Similarity between KO-A4 and KO-A4\_mutated\_20: 75.84\%
Similarity between KO-A5 and KO-A5\_mutated\_1: 75.84\%
Similarity between KO-A5 and KO-A5\_mutated\_2: 75.46\%
Similarity between KO-A5 and KO-A5\_mutated\_3: 74.35\%
Similarity between KO-A5 and KO-A5\_mutated\_4: 75.28\%
Similarity between KO-A5 and KO-A5\_mutated\_5: 75.46\%
Similarity between KO-A5 and KO-A5\_mutated\_6: 74.91\%
Similarity between KO-A5 and KO-A5\_mutated\_7: 76.39\%
Similarity between KO-A5 and KO-A5\_mutated\_8: 75.65\%
Similarity between KO-A5 and KO-A5\_mutated\_9: 75.28\%
Similarity between KO-A5 and KO-A5\_mutated\_10: 75.46\%
Similarity between KO-A5 and KO-A5\_mutated\_11: 74.35\%
Similarity between KO-A5 and KO-A5\_mutated\_12: 75.09\%
Similarity between KO-A5 and KO-A5\_mutated\_13: 76.21\%
Similarity between KO-A5 and KO-A5\_mutated\_14: 74.54\%
Similarity between KO-A5 and KO-A5\_mutated\_15: 73.79\%
Similarity between KO-A5 and KO-A5\_mutated\_16: 74.35\%
Similarity between KO-A5 and KO-A5\_mutated\_17: 73.79\%
Similarity between KO-A5 and KO-A5\_mutated\_18: 75.84\%
Similarity between KO-A5 and KO-A5\_mutated\_19: 74.35\%
Similarity between KO-A5 and KO-A5\_mutated\_20: 75.09\%
Similarity between KO-A6 and KO-A6\_mutated\_1: 73.42\%
Similarity between KO-A6 and KO-A6\_mutated\_2: 74.91\%
Similarity between KO-A6 and KO-A6\_mutated\_3: 74.91\%
Similarity between KO-A6 and KO-A6\_mutated\_4: 75.46\%
Similarity between KO-A6 and KO-A6\_mutated\_5: 74.72\%
Similarity between KO-A6 and KO-A6\_mutated\_6: 75.65\%
Similarity between KO-A6 and KO-A6\_mutated\_7: 76.39\%
Similarity between KO-A6 and KO-A6\_mutated\_8: 74.54\%
Similarity between KO-A6 and KO-A6\_mutated\_9: 75.46\%
Similarity between KO-A6 and KO-A6\_mutated\_10: 74.54\%
Similarity between KO-A6 and KO-A6\_mutated\_11: 73.98\%
Similarity between KO-A6 and KO-A6\_mutated\_12: 74.72\%
Similarity between KO-A6 and KO-A6\_mutated\_13: 75.28\%
Similarity between KO-A6 and KO-A6\_mutated\_14: 74.72\%
Similarity between KO-A6 and KO-A6\_mutated\_15: 76.21\%
Similarity between KO-A6 and KO-A6\_mutated\_16: 75.65\%
Similarity between KO-A6 and KO-A6\_mutated\_17: 75.46\%
Similarity between KO-A6 and KO-A6\_mutated\_18: 75.84\%
Similarity between KO-A6 and KO-A6\_mutated\_19: 75.65\%
Similarity between KO-A6 and KO-A6\_mutated\_20: 73.79\%
Similarity between KO-A7 and KO-A7\_mutated\_1: 75.84\%
Similarity between KO-A7 and KO-A7\_mutated\_2: 76.95\%
Similarity between KO-A7 and KO-A7\_mutated\_3: 74.35\%
Similarity between KO-A7 and KO-A7\_mutated\_4: 75.84\%
Similarity between KO-A7 and KO-A7\_mutated\_5: 75.46\%
Similarity between KO-A7 and KO-A7\_mutated\_6: 75.28\%
Similarity between KO-A7 and KO-A7\_mutated\_7: 75.09\%
Similarity between KO-A7 and KO-A7\_mutated\_8: 73.79\%
Similarity between KO-A7 and KO-A7\_mutated\_9: 74.16\%
Similarity between KO-A7 and KO-A7\_mutated\_10: 76.02\%
Similarity between KO-A7 and KO-A7\_mutated\_11: 75.65\%
Similarity between KO-A7 and KO-A7\_mutated\_12: 74.35\%
Similarity between KO-A7 and KO-A7\_mutated\_13: 75.28\%
Similarity between KO-A7 and KO-A7\_mutated\_14: 76.58\%
Similarity between KO-A7 and KO-A7\_mutated\_15: 74.54\%
Similarity between KO-A7 and KO-A7\_mutated\_16: 74.35\%
Similarity between KO-A7 and KO-A7\_mutated\_17: 75.09\%
Similarity between KO-A7 and KO-A7\_mutated\_18: 76.39\%
Similarity between KO-A7 and KO-A7\_mutated\_19: 76.39\%
Similarity between KO-A7 and KO-A7\_mutated\_20: 75.09\%
Similarity between KO-A8 and KO-A8\_mutated\_1: 74.54\%
Similarity between KO-A8 and KO-A8\_mutated\_2: 75.65\%
Similarity between KO-A8 and KO-A8\_mutated\_3: 74.91\%
Similarity between KO-A8 and KO-A8\_mutated\_4: 73.42\%
Similarity between KO-A8 and KO-A8\_mutated\_5: 76.95\%
Similarity between KO-A8 and KO-A8\_mutated\_6: 76.02\%
Similarity between KO-A8 and KO-A8\_mutated\_7: 74.35\%
Similarity between KO-A8 and KO-A8\_mutated\_8: 76.21\%
Similarity between KO-A8 and KO-A8\_mutated\_9: 76.02\%
Similarity between KO-A8 and KO-A8\_mutated\_10: 74.35\%
Similarity between KO-A8 and KO-A8\_mutated\_11: 73.98\%
Similarity between KO-A8 and KO-A8\_mutated\_12: 76.95\%
Similarity between KO-A8 and KO-A8\_mutated\_13: 75.46\%
Similarity between KO-A8 and KO-A8\_mutated\_14: 76.02\%
Similarity between KO-A8 and KO-A8\_mutated\_15: 75.09\%
Similarity between KO-A8 and KO-A8\_mutated\_16: 75.46\%
Similarity between KO-A8 and KO-A8\_mutated\_17: 74.16\%
Similarity between KO-A8 and KO-A8\_mutated\_18: 74.72\%
Similarity between KO-A8 and KO-A8\_mutated\_19: 74.91\%
Similarity between KO-A8 and KO-A8\_mutated\_20: 76.21\%
Similarity between KO-A9 and KO-A9\_mutated\_1: 74.72\%
Similarity between KO-A9 and KO-A9\_mutated\_2: 76.39\%
Similarity between KO-A9 and KO-A9\_mutated\_3: 75.46\%
Similarity between KO-A9 and KO-A9\_mutated\_4: 74.54\%
Similarity between KO-A9 and KO-A9\_mutated\_5: 75.09\%
Similarity between KO-A9 and KO-A9\_mutated\_6: 76.95\%
Similarity between KO-A9 and KO-A9\_mutated\_7: 74.91\%
Similarity between KO-A9 and KO-A9\_mutated\_8: 75.28\%
Similarity between KO-A9 and KO-A9\_mutated\_9: 75.28\%
Similarity between KO-A9 and KO-A9\_mutated\_10: 75.09\%
Similarity between KO-A9 and KO-A9\_mutated\_11: 74.72\%
Similarity between KO-A9 and KO-A9\_mutated\_12: 74.54\%
Similarity between KO-A9 and KO-A9\_mutated\_13: 74.54\%
Similarity between KO-A9 and KO-A9\_mutated\_14: 74.35\%
Similarity between KO-A9 and KO-A9\_mutated\_15: 74.91\%
Similarity between KO-A9 and KO-A9\_mutated\_16: 74.16\%
Similarity between KO-A9 and KO-A9\_mutated\_17: 73.98\%
Similarity between KO-A9 and KO-A9\_mutated\_18: 74.72\%
Similarity between KO-A9 and KO-A9\_mutated\_19: 74.35\%
Similarity between KO-A9 and KO-A9\_mutated\_20: 74.72\%
Similarity between KO-A10 and KO-A10\_mutated\_1: 75.09\%
Similarity between KO-A10 and KO-A10\_mutated\_2: 74.35\%
Similarity between KO-A10 and KO-A10\_mutated\_3: 74.54\%
Similarity between KO-A10 and KO-A10\_mutated\_4: 75.84\%
Similarity between KO-A10 and KO-A10\_mutated\_5: 75.46\%
Similarity between KO-A10 and KO-A10\_mutated\_6: 74.16\%
Similarity between KO-A10 and KO-A10\_mutated\_7: 74.91\%
Similarity between KO-A10 and KO-A10\_mutated\_8: 74.35\%
Similarity between KO-A10 and KO-A10\_mutated\_9: 74.91\%
Similarity between KO-A10 and KO-A10\_mutated\_10: 75.46\%
Similarity between KO-A10 and KO-A10\_mutated\_11: 75.28\%
Similarity between KO-A10 and KO-A10\_mutated\_12: 75.28\%
Similarity between KO-A10 and KO-A10\_mutated\_13: 76.02\%
Similarity between KO-A10 and KO-A10\_mutated\_14: 74.35\%
Similarity between KO-A10 and KO-A10\_mutated\_15: 73.79\%
Similarity between KO-A10 and KO-A10\_mutated\_16: 73.42\%
Similarity between KO-A10 and KO-A10\_mutated\_17: 74.91\%
Similarity between KO-A10 and KO-A10\_mutated\_18: 75.84\%
Similarity between KO-A10 and KO-A10\_mutated\_19: 75.84\%
Similarity between KO-A10 and KO-A10\_mutated\_20: 76.58\%
Similarity between KO-B1 and KO-B1\_mutated\_1: 73.79\%
Similarity between KO-B1 and KO-B1\_mutated\_2: 74.91\%
Similarity between KO-B1 and KO-B1\_mutated\_3: 74.91\%
Similarity between KO-B1 and KO-B1\_mutated\_4: 75.46\%
Similarity between KO-B1 and KO-B1\_mutated\_5: 74.72\%
Similarity between KO-B1 and KO-B1\_mutated\_6: 74.91\%
Similarity between KO-B1 and KO-B1\_mutated\_7: 75.28\%
Similarity between KO-B1 and KO-B1\_mutated\_8: 75.84\%
Similarity between KO-B1 and KO-B1\_mutated\_9: 73.79\%
Similarity between KO-B1 and KO-B1\_mutated\_10: 76.58\%
Similarity between KO-B1 and KO-B1\_mutated\_11: 76.02\%
Similarity between KO-B1 and KO-B1\_mutated\_12: 75.09\%
Similarity between KO-B1 and KO-B1\_mutated\_13: 75.65\%
Similarity between KO-B1 and KO-B1\_mutated\_14: 75.28\%
Similarity between KO-B1 and KO-B1\_mutated\_15: 74.54\%
Similarity between KO-B1 and KO-B1\_mutated\_16: 76.39\%
Similarity between KO-B1 and KO-B1\_mutated\_17: 73.42\%
Similarity between KO-B1 and KO-B1\_mutated\_18: 75.65\%
Similarity between KO-B1 and KO-B1\_mutated\_19: 74.35\%
Similarity between KO-B1 and KO-B1\_mutated\_20: 75.84\%
Similarity between KO-B2 and KO-B2\_mutated\_1: 75.65\%
Similarity between KO-B2 and KO-B2\_mutated\_2: 75.28\%
Similarity between KO-B2 and KO-B2\_mutated\_3: 75.84\%
Similarity between KO-B2 and KO-B2\_mutated\_4: 75.09\%
Similarity between KO-B2 and KO-B2\_mutated\_5: 76.77\%
Similarity between KO-B2 and KO-B2\_mutated\_6: 75.09\%
Similarity between KO-B2 and KO-B2\_mutated\_7: 74.54\%
Similarity between KO-B2 and KO-B2\_mutated\_8: 74.72\%
Similarity between KO-B2 and KO-B2\_mutated\_9: 75.09\%
Similarity between KO-B2 and KO-B2\_mutated\_10: 75.09\%
Similarity between KO-B2 and KO-B2\_mutated\_11: 75.65\%
Similarity between KO-B2 and KO-B2\_mutated\_12: 75.84\%
Similarity between KO-B2 and KO-B2\_mutated\_13: 74.91\%
Similarity between KO-B2 and KO-B2\_mutated\_14: 75.84\%
Similarity between KO-B2 and KO-B2\_mutated\_15: 74.91\%
Similarity between KO-B2 and KO-B2\_mutated\_16: 75.65\%
Similarity between KO-B2 and KO-B2\_mutated\_17: 73.79\%
Similarity between KO-B2 and KO-B2\_mutated\_18: 76.02\%
Similarity between KO-B2 and KO-B2\_mutated\_19: 76.21\%
Similarity between KO-B2 and KO-B2\_mutated\_20: 75.28\%
Similarity between KO-B3 and KO-B3\_mutated\_1: 75.28\%
Similarity between KO-B3 and KO-B3\_mutated\_2: 75.09\%
Similarity between KO-B3 and KO-B3\_mutated\_3: 74.54\%
Similarity between KO-B3 and KO-B3\_mutated\_4: 74.35\%
Similarity between KO-B3 and KO-B3\_mutated\_5: 75.09\%
Similarity between KO-B3 and KO-B3\_mutated\_6: 76.77\%
Similarity between KO-B3 and KO-B3\_mutated\_7: 75.46\%
Similarity between KO-B3 and KO-B3\_mutated\_8: 74.91\%
Similarity between KO-B3 and KO-B3\_mutated\_9: 76.02\%
Similarity between KO-B3 and KO-B3\_mutated\_10: 74.54\%
Similarity between KO-B3 and KO-B3\_mutated\_11: 75.28\%
Similarity between KO-B3 and KO-B3\_mutated\_12: 75.65\%
Similarity between KO-B3 and KO-B3\_mutated\_13: 75.28\%
Similarity between KO-B3 and KO-B3\_mutated\_14: 74.54\%
Similarity between KO-B3 and KO-B3\_mutated\_15: 75.28\%
Similarity between KO-B3 and KO-B3\_mutated\_16: 74.91\%
Similarity between KO-B3 and KO-B3\_mutated\_17: 73.61\%
Similarity between KO-B3 and KO-B3\_mutated\_18: 74.72\%
Similarity between KO-B3 and KO-B3\_mutated\_19: 76.58\%
Similarity between KO-B3 and KO-B3\_mutated\_20: 74.72\%
Similarity between KO-B4 and KO-B4\_mutated\_1: 75.28\%
Similarity between KO-B4 and KO-B4\_mutated\_2: 74.16\%
Similarity between KO-B4 and KO-B4\_mutated\_3: 75.84\%
Similarity between KO-B4 and KO-B4\_mutated\_4: 75.09\%
Similarity between KO-B4 and KO-B4\_mutated\_5: 74.91\%
Similarity between KO-B4 and KO-B4\_mutated\_6: 73.98\%
Similarity between KO-B4 and KO-B4\_mutated\_7: 74.16\%
Similarity between KO-B4 and KO-B4\_mutated\_8: 74.54\%
Similarity between KO-B4 and KO-B4\_mutated\_9: 73.79\%
Similarity between KO-B4 and KO-B4\_mutated\_10: 75.28\%
Similarity between KO-B4 and KO-B4\_mutated\_11: 75.65\%
Similarity between KO-B4 and KO-B4\_mutated\_12: 75.65\%
Similarity between KO-B4 and KO-B4\_mutated\_13: 74.91\%
Similarity between KO-B4 and KO-B4\_mutated\_14: 75.28\%
Similarity between KO-B4 and KO-B4\_mutated\_15: 74.16\%
Similarity between KO-B4 and KO-B4\_mutated\_16: 74.91\%
Similarity between KO-B4 and KO-B4\_mutated\_17: 74.72\%
Similarity between KO-B4 and KO-B4\_mutated\_18: 75.46\%
Similarity between KO-B4 and KO-B4\_mutated\_19: 74.54\%
Similarity between KO-B4 and KO-B4\_mutated\_20: 74.54\%
Similarity between KO-B5 and KO-B5\_mutated\_1: 75.65\%
Similarity between KO-B5 and KO-B5\_mutated\_2: 74.54\%
Similarity between KO-B5 and KO-B5\_mutated\_3: 74.54\%
Similarity between KO-B5 and KO-B5\_mutated\_4: 74.54\%
Similarity between KO-B5 and KO-B5\_mutated\_5: 75.46\%
Similarity between KO-B5 and KO-B5\_mutated\_6: 76.39\%
Similarity between KO-B5 and KO-B5\_mutated\_7: 74.72\%
Similarity between KO-B5 and KO-B5\_mutated\_8: 74.72\%
Similarity between KO-B5 and KO-B5\_mutated\_9: 74.54\%
Similarity between KO-B5 and KO-B5\_mutated\_10: 73.61\%
Similarity between KO-B5 and KO-B5\_mutated\_11: 74.35\%
Similarity between KO-B5 and KO-B5\_mutated\_12: 75.46\%
Similarity between KO-B5 and KO-B5\_mutated\_13: 76.21\%
Similarity between KO-B5 and KO-B5\_mutated\_14: 74.54\%
Similarity between KO-B5 and KO-B5\_mutated\_15: 74.91\%
Similarity between KO-B5 and KO-B5\_mutated\_16: 74.72\%
Similarity between KO-B5 and KO-B5\_mutated\_17: 74.54\%
Similarity between KO-B5 and KO-B5\_mutated\_18: 74.35\%
Similarity between KO-B5 and KO-B5\_mutated\_19: 73.79\%
Similarity between KO-B5 and KO-B5\_mutated\_20: 74.16\%
Similarity between KO-B6 and KO-B6\_mutated\_1: 74.35\%
Similarity between KO-B6 and KO-B6\_mutated\_2: 75.46\%
Similarity between KO-B6 and KO-B6\_mutated\_3: 75.28\%
Similarity between KO-B6 and KO-B6\_mutated\_4: 74.91\%
Similarity between KO-B6 and KO-B6\_mutated\_5: 74.16\%
Similarity between KO-B6 and KO-B6\_mutated\_6: 74.54\%
Similarity between KO-B6 and KO-B6\_mutated\_7: 74.54\%
Similarity between KO-B6 and KO-B6\_mutated\_8: 75.46\%
Similarity between KO-B6 and KO-B6\_mutated\_9: 75.09\%
Similarity between KO-B6 and KO-B6\_mutated\_10: 74.35\%
Similarity between KO-B6 and KO-B6\_mutated\_11: 75.65\%
Similarity between KO-B6 and KO-B6\_mutated\_12: 73.79\%
Similarity between KO-B6 and KO-B6\_mutated\_13: 74.16\%
Similarity between KO-B6 and KO-B6\_mutated\_14: 74.35\%
Similarity between KO-B6 and KO-B6\_mutated\_15: 74.72\%
Similarity between KO-B6 and KO-B6\_mutated\_16: 75.46\%
Similarity between KO-B6 and KO-B6\_mutated\_17: 76.39\%
Similarity between KO-B6 and KO-B6\_mutated\_18: 76.02\%
Similarity between KO-B6 and KO-B6\_mutated\_19: 76.02\%
Similarity between KO-B6 and KO-B6\_mutated\_20: 75.84\%
Similarity between KO-B7 and KO-B7\_mutated\_1: 74.54\%
Similarity between KO-B7 and KO-B7\_mutated\_2: 76.58\%
Similarity between KO-B7 and KO-B7\_mutated\_3: 76.39\%
Similarity between KO-B7 and KO-B7\_mutated\_4: 76.02\%
Similarity between KO-B7 and KO-B7\_mutated\_5: 75.84\%
Similarity between KO-B7 and KO-B7\_mutated\_6: 73.61\%
Similarity between KO-B7 and KO-B7\_mutated\_7: 75.65\%
Similarity between KO-B7 and KO-B7\_mutated\_8: 74.91\%
Similarity between KO-B7 and KO-B7\_mutated\_9: 73.98\%
Similarity between KO-B7 and KO-B7\_mutated\_10: 74.91\%
Similarity between KO-B7 and KO-B7\_mutated\_11: 75.46\%
Similarity between KO-B7 and KO-B7\_mutated\_12: 74.16\%
Similarity between KO-B7 and KO-B7\_mutated\_13: 75.65\%
Similarity between KO-B7 and KO-B7\_mutated\_14: 75.84\%
Similarity between KO-B7 and KO-B7\_mutated\_15: 74.72\%
Similarity between KO-B7 and KO-B7\_mutated\_16: 75.46\%
Similarity between KO-B7 and KO-B7\_mutated\_17: 75.09\%
Similarity between KO-B7 and KO-B7\_mutated\_18: 74.91\%
Similarity between KO-B7 and KO-B7\_mutated\_19: 73.23\%
Similarity between KO-B7 and KO-B7\_mutated\_20: 75.46\%
Similarity between KO-B8 and KO-B8\_mutated\_1: 74.91\%
Similarity between KO-B8 and KO-B8\_mutated\_2: 75.09\%
Similarity between KO-B8 and KO-B8\_mutated\_3: 75.65\%
Similarity between KO-B8 and KO-B8\_mutated\_4: 76.02\%
Similarity between KO-B8 and KO-B8\_mutated\_5: 75.84\%
Similarity between KO-B8 and KO-B8\_mutated\_6: 75.46\%
Similarity between KO-B8 and KO-B8\_mutated\_7: 74.91\%
Similarity between KO-B8 and KO-B8\_mutated\_8: 75.65\%
Similarity between KO-B8 and KO-B8\_mutated\_9: 74.54\%
Similarity between KO-B8 and KO-B8\_mutated\_10: 76.02\%
Similarity between KO-B8 and KO-B8\_mutated\_11: 75.28\%
Similarity between KO-B8 and KO-B8\_mutated\_12: 75.46\%
Similarity between KO-B8 and KO-B8\_mutated\_13: 75.84\%
Similarity between KO-B8 and KO-B8\_mutated\_14: 77.32\%
Similarity between KO-B8 and KO-B8\_mutated\_15: 75.84\%
Similarity between KO-B8 and KO-B8\_mutated\_16: 75.46\%
Similarity between KO-B8 and KO-B8\_mutated\_17: 75.09\%
Similarity between KO-B8 and KO-B8\_mutated\_18: 74.54\%
Similarity between KO-B8 and KO-B8\_mutated\_19: 73.98\%
Similarity between KO-B8 and KO-B8\_mutated\_20: 77.51\%
Similarity between KO-B9 and KO-B9\_mutated\_1: 73.79\%
Similarity between KO-B9 and KO-B9\_mutated\_2: 75.46\%
Similarity between KO-B9 and KO-B9\_mutated\_3: 73.23\%
Similarity between KO-B9 and KO-B9\_mutated\_4: 75.09\%
Similarity between KO-B9 and KO-B9\_mutated\_5: 74.16\%
Similarity between KO-B9 and KO-B9\_mutated\_6: 74.54\%
Similarity between KO-B9 and KO-B9\_mutated\_7: 75.84\%
Similarity between KO-B9 and KO-B9\_mutated\_8: 76.02\%
Similarity between KO-B9 and KO-B9\_mutated\_9: 75.84\%
Similarity between KO-B9 and KO-B9\_mutated\_10: 75.84\%
Similarity between KO-B9 and KO-B9\_mutated\_11: 75.84\%
Similarity between KO-B9 and KO-B9\_mutated\_12: 75.84\%
Similarity between KO-B9 and KO-B9\_mutated\_13: 75.65\%
Similarity between KO-B9 and KO-B9\_mutated\_14: 75.09\%
Similarity between KO-B9 and KO-B9\_mutated\_15: 76.39\%
Similarity between KO-B9 and KO-B9\_mutated\_16: 75.09\%
Similarity between KO-B9 and KO-B9\_mutated\_17: 74.54\%
Similarity between KO-B9 and KO-B9\_mutated\_18: 75.65\%
Similarity between KO-B9 and KO-B9\_mutated\_19: 74.91\%
Similarity between KO-B9 and KO-B9\_mutated\_20: 73.79\%
Similarity between KO-B10 and KO-B10\_mutated\_1: 74.54\%
Similarity between KO-B10 and KO-B10\_mutated\_2: 75.09\%
Similarity between KO-B10 and KO-B10\_mutated\_3: 75.46\%
Similarity between KO-B10 and KO-B10\_mutated\_4: 73.98\%
Similarity between KO-B10 and KO-B10\_mutated\_5: 75.46\%
Similarity between KO-B10 and KO-B10\_mutated\_6: 75.28\%
Similarity between KO-B10 and KO-B10\_mutated\_7: 74.16\%
Similarity between KO-B10 and KO-B10\_mutated\_8: 76.21\%
Similarity between KO-B10 and KO-B10\_mutated\_9: 76.02\%
Similarity between KO-B10 and KO-B10\_mutated\_10: 75.46\%
Similarity between KO-B10 and KO-B10\_mutated\_11: 76.58\%
Similarity between KO-B10 and KO-B10\_mutated\_12: 76.39\%
Similarity between KO-B10 and KO-B10\_mutated\_13: 74.91\%
Similarity between KO-B10 and KO-B10\_mutated\_14: 76.77\%
Similarity between KO-B10 and KO-B10\_mutated\_15: 75.09\%
Similarity between KO-B10 and KO-B10\_mutated\_16: 75.28\%
Similarity between KO-B10 and KO-B10\_mutated\_17: 76.02\%
Similarity between KO-B10 and KO-B10\_mutated\_18: 75.28\%
Similarity between KO-B10 and KO-B10\_mutated\_19: 74.72\%
Similarity between KO-B10 and KO-B10\_mutated\_20: 75.46\%
Similarity between KO-J1 and KO-J1\_mutated\_1: 74.91\%
Similarity between KO-J1 and KO-J1\_mutated\_2: 74.16\%
Similarity between KO-J1 and KO-J1\_mutated\_3: 74.35\%
Similarity between KO-J1 and KO-J1\_mutated\_4: 73.98\%
Similarity between KO-J1 and KO-J1\_mutated\_5: 75.46\%
Similarity between KO-J1 and KO-J1\_mutated\_6: 75.84\%
Similarity between KO-J1 and KO-J1\_mutated\_7: 73.98\%
Similarity between KO-J1 and KO-J1\_mutated\_8: 73.61\%
Similarity between KO-J1 and KO-J1\_mutated\_9: 75.84\%
Similarity between KO-J1 and KO-J1\_mutated\_10: 75.09\%
Similarity between KO-J1 and KO-J1\_mutated\_11: 73.98\%
Similarity between KO-J1 and KO-J1\_mutated\_12: 74.54\%
Similarity between KO-J1 and KO-J1\_mutated\_13: 75.09\%
Similarity between KO-J1 and KO-J1\_mutated\_14: 74.54\%
Similarity between KO-J1 and KO-J1\_mutated\_15: 76.21\%
Similarity between KO-J1 and KO-J1\_mutated\_16: 74.16\%
Similarity between KO-J1 and KO-J1\_mutated\_17: 74.54\%
Similarity between KO-J1 and KO-J1\_mutated\_18: 75.46\%
Similarity between KO-J1 and KO-J1\_mutated\_19: 74.16\%
Similarity between KO-J1 and KO-J1\_mutated\_20: 74.91\%
Similarity between KO-J2 and KO-J2\_mutated\_1: 76.95\%
Similarity between KO-J2 and KO-J2\_mutated\_2: 76.39\%
Similarity between KO-J2 and KO-J2\_mutated\_3: 75.28\%
Similarity between KO-J2 and KO-J2\_mutated\_4: 74.91\%
Similarity between KO-J2 and KO-J2\_mutated\_5: 76.39\%
Similarity between KO-J2 and KO-J2\_mutated\_6: 73.61\%
Similarity between KO-J2 and KO-J2\_mutated\_7: 75.84\%
Similarity between KO-J2 and KO-J2\_mutated\_8: 74.54\%
Similarity between KO-J2 and KO-J2\_mutated\_9: 74.91\%
Similarity between KO-J2 and KO-J2\_mutated\_10: 74.54\%
Similarity between KO-J2 and KO-J2\_mutated\_11: 74.91\%
Similarity between KO-J2 and KO-J2\_mutated\_12: 74.16\%
Similarity between KO-J2 and KO-J2\_mutated\_13: 75.28\%
Similarity between KO-J2 and KO-J2\_mutated\_14: 73.79\%
Similarity between KO-J2 and KO-J2\_mutated\_15: 74.16\%
Similarity between KO-J2 and KO-J2\_mutated\_16: 75.09\%
Similarity between KO-J2 and KO-J2\_mutated\_17: 76.02\%
Similarity between KO-J2 and KO-J2\_mutated\_18: 76.21\%
Similarity between KO-J2 and KO-J2\_mutated\_19: 76.02\%
Similarity between KO-J2 and KO-J2\_mutated\_20: 76.21\%
Similarity between KO-J3 and KO-J3\_mutated\_1: 75.09\%
Similarity between KO-J3 and KO-J3\_mutated\_2: 74.35\%
Similarity between KO-J3 and KO-J3\_mutated\_3: 75.46\%
Similarity between KO-J3 and KO-J3\_mutated\_4: 75.46\%
Similarity between KO-J3 and KO-J3\_mutated\_5: 75.65\%
Similarity between KO-J3 and KO-J3\_mutated\_6: 74.72\%
Similarity between KO-J3 and KO-J3\_mutated\_7: 74.91\%
Similarity between KO-J3 and KO-J3\_mutated\_8: 75.09\%
Similarity between KO-J3 and KO-J3\_mutated\_9: 75.28\%
Similarity between KO-J3 and KO-J3\_mutated\_10: 74.54\%
Similarity between KO-J3 and KO-J3\_mutated\_11: 75.84\%
Similarity between KO-J3 and KO-J3\_mutated\_12: 76.39\%
Similarity between KO-J3 and KO-J3\_mutated\_13: 75.84\%
Similarity between KO-J3 and KO-J3\_mutated\_14: 74.54\%
Similarity between KO-J3 and KO-J3\_mutated\_15: 73.42\%
Similarity between KO-J3 and KO-J3\_mutated\_16: 76.77\%
Similarity between KO-J3 and KO-J3\_mutated\_17: 75.84\%
Similarity between KO-J3 and KO-J3\_mutated\_18: 75.09\%
Similarity between KO-J3 and KO-J3\_mutated\_19: 74.91\%
Similarity between KO-J3 and KO-J3\_mutated\_20: 73.61\%
Similarity between KO-J4 and KO-J4\_mutated\_1: 75.28\%
Similarity between KO-J4 and KO-J4\_mutated\_2: 75.84\%
Similarity between KO-J4 and KO-J4\_mutated\_3: 74.91\%
Similarity between KO-J4 and KO-J4\_mutated\_4: 74.91\%
Similarity between KO-J4 and KO-J4\_mutated\_5: 75.46\%
Similarity between KO-J4 and KO-J4\_mutated\_6: 75.46\%
Similarity between KO-J4 and KO-J4\_mutated\_7: 74.72\%
Similarity between KO-J4 and KO-J4\_mutated\_8: 74.35\%
Similarity between KO-J4 and KO-J4\_mutated\_9: 75.28\%
Similarity between KO-J4 and KO-J4\_mutated\_10: 74.72\%
Similarity between KO-J4 and KO-J4\_mutated\_11: 74.54\%
Similarity between KO-J4 and KO-J4\_mutated\_12: 73.79\%
Similarity between KO-J4 and KO-J4\_mutated\_13: 74.35\%
Similarity between KO-J4 and KO-J4\_mutated\_14: 77.32\%
Similarity between KO-J4 and KO-J4\_mutated\_15: 75.46\%
Similarity between KO-J4 and KO-J4\_mutated\_16: 76.02\%
Similarity between KO-J4 and KO-J4\_mutated\_17: 74.72\%
Similarity between KO-J4 and KO-J4\_mutated\_18: 75.09\%
Similarity between KO-J4 and KO-J4\_mutated\_19: 73.98\%
Similarity between KO-J4 and KO-J4\_mutated\_20: 74.91\%
Similarity between KO-J5 and KO-J5\_mutated\_1: 76.58\%
Similarity between KO-J5 and KO-J5\_mutated\_2: 76.02\%
Similarity between KO-J5 and KO-J5\_mutated\_3: 75.65\%
Similarity between KO-J5 and KO-J5\_mutated\_4: 75.28\%
Similarity between KO-J5 and KO-J5\_mutated\_5: 75.28\%
Similarity between KO-J5 and KO-J5\_mutated\_6: 74.35\%
Similarity between KO-J5 and KO-J5\_mutated\_7: 75.84\%
Similarity between KO-J5 and KO-J5\_mutated\_8: 73.61\%
Similarity between KO-J5 and KO-J5\_mutated\_9: 74.72\%
Similarity between KO-J5 and KO-J5\_mutated\_10: 74.54\%
Similarity between KO-J5 and KO-J5\_mutated\_11: 76.21\%
Similarity between KO-J5 and KO-J5\_mutated\_12: 74.91\%
Similarity between KO-J5 and KO-J5\_mutated\_13: 74.54\%
Similarity between KO-J5 and KO-J5\_mutated\_14: 76.02\%
Similarity between KO-J5 and KO-J5\_mutated\_15: 74.16\%
Similarity between KO-J5 and KO-J5\_mutated\_16: 74.35\%
Similarity between KO-J5 and KO-J5\_mutated\_17: 75.28\%
Similarity between KO-J5 and KO-J5\_mutated\_18: 75.09\%
Similarity between KO-J5 and KO-J5\_mutated\_19: 74.54\%
Similarity between KO-J5 and KO-J5\_mutated\_20: 74.91\%
Similarity between KO-J6 and KO-J6\_mutated\_1: 74.91\%
Similarity between KO-J6 and KO-J6\_mutated\_2: 74.35\%
Similarity between KO-J6 and KO-J6\_mutated\_3: 74.91\%
Similarity between KO-J6 and KO-J6\_mutated\_4: 74.54\%
Similarity between KO-J6 and KO-J6\_mutated\_5: 75.28\%
Similarity between KO-J6 and KO-J6\_mutated\_6: 74.35\%
Similarity between KO-J6 and KO-J6\_mutated\_7: 75.28\%
Similarity between KO-J6 and KO-J6\_mutated\_8: 75.65\%
Similarity between KO-J6 and KO-J6\_mutated\_9: 74.72\%
Similarity between KO-J6 and KO-J6\_mutated\_10: 76.39\%
Similarity between KO-J6 and KO-J6\_mutated\_11: 74.16\%
Similarity between KO-J6 and KO-J6\_mutated\_12: 73.98\%
Similarity between KO-J6 and KO-J6\_mutated\_13: 75.28\%
Similarity between KO-J6 and KO-J6\_mutated\_14: 75.46\%
Similarity between KO-J6 and KO-J6\_mutated\_15: 75.09\%
Similarity between KO-J6 and KO-J6\_mutated\_16: 74.54\%
Similarity between KO-J6 and KO-J6\_mutated\_17: 77.51\%
Similarity between KO-J6 and KO-J6\_mutated\_18: 74.16\%
Similarity between KO-J6 and KO-J6\_mutated\_19: 74.72\%
Similarity between KO-J6 and KO-J6\_mutated\_20: 76.77\%
Similarity between KO-J7 and KO-J7\_mutated\_1: 75.28\%
Similarity between KO-J7 and KO-J7\_mutated\_2: 74.91\%
Similarity between KO-J7 and KO-J7\_mutated\_3: 76.95\%
Similarity between KO-J7 and KO-J7\_mutated\_4: 74.16\%
Similarity between KO-J7 and KO-J7\_mutated\_5: 76.95\%
Similarity between KO-J7 and KO-J7\_mutated\_6: 76.77\%
Similarity between KO-J7 and KO-J7\_mutated\_7: 74.54\%
Similarity between KO-J7 and KO-J7\_mutated\_8: 74.91\%
Similarity between KO-J7 and KO-J7\_mutated\_9: 74.72\%
Similarity between KO-J7 and KO-J7\_mutated\_10: 74.16\%
Similarity between KO-J7 and KO-J7\_mutated\_11: 75.65\%
Similarity between KO-J7 and KO-J7\_mutated\_12: 75.84\%
Similarity between KO-J7 and KO-J7\_mutated\_13: 75.28\%
Similarity between KO-J7 and KO-J7\_mutated\_14: 75.09\%
Similarity between KO-J7 and KO-J7\_mutated\_15: 75.28\%
Similarity between KO-J7 and KO-J7\_mutated\_16: 76.02\%
Similarity between KO-J7 and KO-J7\_mutated\_17: 75.09\%
Similarity between KO-J7 and KO-J7\_mutated\_18: 74.72\%
Similarity between KO-J7 and KO-J7\_mutated\_19: 75.09\%
Similarity between KO-J7 and KO-J7\_mutated\_20: 75.46\%
Similarity between KO-J8 and KO-J8\_mutated\_1: 75.09\%
Similarity between KO-J8 and KO-J8\_mutated\_2: 74.16\%
Similarity between KO-J8 and KO-J8\_mutated\_3: 76.02\%
Similarity between KO-J8 and KO-J8\_mutated\_4: 75.09\%
Similarity between KO-J8 and KO-J8\_mutated\_5: 74.72\%
Similarity between KO-J8 and KO-J8\_mutated\_6: 75.46\%
Similarity between KO-J8 and KO-J8\_mutated\_7: 73.98\%
Similarity between KO-J8 and KO-J8\_mutated\_8: 74.16\%
Similarity between KO-J8 and KO-J8\_mutated\_9: 74.91\%
Similarity between KO-J8 and KO-J8\_mutated\_10: 76.02\%
Similarity between KO-J8 and KO-J8\_mutated\_11: 75.28\%
Similarity between KO-J8 and KO-J8\_mutated\_12: 75.46\%
Similarity between KO-J8 and KO-J8\_mutated\_13: 74.35\%
Similarity between KO-J8 and KO-J8\_mutated\_14: 75.65\%
Similarity between KO-J8 and KO-J8\_mutated\_15: 76.02\%
Similarity between KO-J8 and KO-J8\_mutated\_16: 75.46\%
Similarity between KO-J8 and KO-J8\_mutated\_17: 76.02\%
Similarity between KO-J8 and KO-J8\_mutated\_18: 75.65\%
Similarity between KO-J8 and KO-J8\_mutated\_19: 74.35\%
Similarity between KO-J8 and KO-J8\_mutated\_20: 75.46\%
Similarity between KO-J9 and KO-J9\_mutated\_1: 75.28\%
Similarity between KO-J9 and KO-J9\_mutated\_2: 74.72\%
Similarity between KO-J9 and KO-J9\_mutated\_3: 75.09\%
Similarity between KO-J9 and KO-J9\_mutated\_4: 75.28\%
Similarity between KO-J9 and KO-J9\_mutated\_5: 74.72\%
Similarity between KO-J9 and KO-J9\_mutated\_6: 75.28\%
Similarity between KO-J9 and KO-J9\_mutated\_7: 75.09\%
Similarity between KO-J9 and KO-J9\_mutated\_8: 74.91\%
Similarity between KO-J9 and KO-J9\_mutated\_9: 74.16\%
Similarity between KO-J9 and KO-J9\_mutated\_10: 74.54\%
Similarity between KO-J9 and KO-J9\_mutated\_11: 74.16\%
Similarity between KO-J9 and KO-J9\_mutated\_12: 73.79\%
Similarity between KO-J9 and KO-J9\_mutated\_13: 75.28\%
Similarity between KO-J9 and KO-J9\_mutated\_14: 75.84\%
Similarity between KO-J9 and KO-J9\_mutated\_15: 75.46\%
Similarity between KO-J9 and KO-J9\_mutated\_16: 76.02\%
Similarity between KO-J9 and KO-J9\_mutated\_17: 75.28\%
Similarity between KO-J9 and KO-J9\_mutated\_18: 75.65\%
Similarity between KO-J9 and KO-J9\_mutated\_19: 75.84\%
Similarity between KO-J9 and KO-J9\_mutated\_20: 75.28\%
Similarity between KO-J10 and KO-J10\_mutated\_1: 76.02\%
Similarity between KO-J10 and KO-J10\_mutated\_2: 75.46\%
Similarity between KO-J10 and KO-J10\_mutated\_3: 76.02\%
Similarity between KO-J10 and KO-J10\_mutated\_4: 74.54\%
Similarity between KO-J10 and KO-J10\_mutated\_5: 76.39\%
Similarity between KO-J10 and KO-J10\_mutated\_6: 74.54\%
Similarity between KO-J10 and KO-J10\_mutated\_7: 74.72\%
Similarity between KO-J10 and KO-J10\_mutated\_8: 74.91\%
Similarity between KO-J10 and KO-J10\_mutated\_9: 74.54\%
Similarity between KO-J10 and KO-J10\_mutated\_10: 76.21\%
Similarity between KO-J10 and KO-J10\_mutated\_11: 75.84\%
Similarity between KO-J10 and KO-J10\_mutated\_12: 74.91\%
Similarity between KO-J10 and KO-J10\_mutated\_13: 75.46\%
Similarity between KO-J10 and KO-J10\_mutated\_14: 74.16\%
Similarity between KO-J10 and KO-J10\_mutated\_15: 75.28\%
Similarity between KO-J10 and KO-J10\_mutated\_16: 74.54\%
Similarity between KO-J10 and KO-J10\_mutated\_17: 74.91\%
Similarity between KO-J10 and KO-J10\_mutated\_18: 76.21\%
Similarity between KO-J10 and KO-J10\_mutated\_19: 75.84\%
Similarity between KO-J10 and KO-J10\_mutated\_20: 75.65\%
Similarity between KO-C1 and KO-C1\_mutated\_1: 75.09\%
Similarity between KO-C1 and KO-C1\_mutated\_2: 76.58\%
Similarity between KO-C1 and KO-C1\_mutated\_3: 75.09\%
Similarity between KO-C1 and KO-C1\_mutated\_4: 75.65\%
Similarity between KO-C1 and KO-C1\_mutated\_5: 75.46\%
Similarity between KO-C1 and KO-C1\_mutated\_6: 76.02\%
Similarity between KO-C1 and KO-C1\_mutated\_7: 73.23\%
Similarity between KO-C1 and KO-C1\_mutated\_8: 75.46\%
Similarity between KO-C1 and KO-C1\_mutated\_9: 74.72\%
Similarity between KO-C1 and KO-C1\_mutated\_10: 73.98\%
Similarity between KO-C1 and KO-C1\_mutated\_11: 75.65\%
Similarity between KO-C1 and KO-C1\_mutated\_12: 74.72\%
Similarity between KO-C1 and KO-C1\_mutated\_13: 75.65\%
Similarity between KO-C1 and KO-C1\_mutated\_14: 76.21\%
Similarity between KO-C1 and KO-C1\_mutated\_15: 74.35\%
Similarity between KO-C1 and KO-C1\_mutated\_16: 74.54\%
Similarity between KO-C1 and KO-C1\_mutated\_17: 73.98\%
Similarity between KO-C1 and KO-C1\_mutated\_18: 75.46\%
Similarity between KO-C1 and KO-C1\_mutated\_19: 75.65\%
Similarity between KO-C1 and KO-C1\_mutated\_20: 75.84\%
Similarity between KO-C2 and KO-C2\_mutated\_1: 75.28\%
Similarity between KO-C2 and KO-C2\_mutated\_2: 75.46\%
Similarity between KO-C2 and KO-C2\_mutated\_3: 75.28\%
Similarity between KO-C2 and KO-C2\_mutated\_4: 74.54\%
Similarity between KO-C2 and KO-C2\_mutated\_5: 75.09\%
Similarity between KO-C2 and KO-C2\_mutated\_6: 75.65\%
Similarity between KO-C2 and KO-C2\_mutated\_7: 75.09\%
Similarity between KO-C2 and KO-C2\_mutated\_8: 76.77\%
Similarity between KO-C2 and KO-C2\_mutated\_9: 75.46\%
Similarity between KO-C2 and KO-C2\_mutated\_10: 76.58\%
Similarity between KO-C2 and KO-C2\_mutated\_11: 73.98\%
Similarity between KO-C2 and KO-C2\_mutated\_12: 75.65\%
Similarity between KO-C2 and KO-C2\_mutated\_13: 74.35\%
Similarity between KO-C2 and KO-C2\_mutated\_14: 74.91\%
Similarity between KO-C2 and KO-C2\_mutated\_15: 75.46\%
Similarity between KO-C2 and KO-C2\_mutated\_16: 75.65\%
Similarity between KO-C2 and KO-C2\_mutated\_17: 74.54\%
Similarity between KO-C2 and KO-C2\_mutated\_18: 75.09\%
Similarity between KO-C2 and KO-C2\_mutated\_19: 75.28\%
Similarity between KO-C2 and KO-C2\_mutated\_20: 76.95\%
Similarity between KO-C3 and KO-C3\_mutated\_1: 74.35\%
Similarity between KO-C3 and KO-C3\_mutated\_2: 76.95\%
Similarity between KO-C3 and KO-C3\_mutated\_3: 76.02\%
Similarity between KO-C3 and KO-C3\_mutated\_4: 74.72\%
Similarity between KO-C3 and KO-C3\_mutated\_5: 74.72\%
Similarity between KO-C3 and KO-C3\_mutated\_6: 74.35\%
Similarity between KO-C3 and KO-C3\_mutated\_7: 76.21\%
Similarity between KO-C3 and KO-C3\_mutated\_8: 74.72\%
Similarity between KO-C3 and KO-C3\_mutated\_9: 74.54\%
Similarity between KO-C3 and KO-C3\_mutated\_10: 75.09\%
Similarity between KO-C3 and KO-C3\_mutated\_11: 75.28\%
Similarity between KO-C3 and KO-C3\_mutated\_12: 75.09\%
Similarity between KO-C3 and KO-C3\_mutated\_13: 76.02\%
Similarity between KO-C3 and KO-C3\_mutated\_14: 74.35\%
Similarity between KO-C3 and KO-C3\_mutated\_15: 74.72\%
Similarity between KO-C3 and KO-C3\_mutated\_16: 76.39\%
Similarity between KO-C3 and KO-C3\_mutated\_17: 75.84\%
Similarity between KO-C3 and KO-C3\_mutated\_18: 74.54\%
Similarity between KO-C3 and KO-C3\_mutated\_19: 76.02\%
Similarity between KO-C3 and KO-C3\_mutated\_20: 75.46\%
Similarity between KO-C4 and KO-C4\_mutated\_1: 74.91\%
Similarity between KO-C4 and KO-C4\_mutated\_2: 74.91\%
Similarity between KO-C4 and KO-C4\_mutated\_3: 76.77\%
Similarity between KO-C4 and KO-C4\_mutated\_4: 74.72\%
Similarity between KO-C4 and KO-C4\_mutated\_5: 75.09\%
Similarity between KO-C4 and KO-C4\_mutated\_6: 76.02\%
Similarity between KO-C4 and KO-C4\_mutated\_7: 75.65\%
Similarity between KO-C4 and KO-C4\_mutated\_8: 74.35\%
Similarity between KO-C4 and KO-C4\_mutated\_9: 74.91\%
Similarity between KO-C4 and KO-C4\_mutated\_10: 75.28\%
Similarity between KO-C4 and KO-C4\_mutated\_11: 75.65\%
Similarity between KO-C4 and KO-C4\_mutated\_12: 75.84\%
Similarity between KO-C4 and KO-C4\_mutated\_13: 75.46\%
Similarity between KO-C4 and KO-C4\_mutated\_14: 74.16\%
Similarity between KO-C4 and KO-C4\_mutated\_15: 75.65\%
Similarity between KO-C4 and KO-C4\_mutated\_16: 75.09\%
Similarity between KO-C4 and KO-C4\_mutated\_17: 75.65\%
Similarity between KO-C4 and KO-C4\_mutated\_18: 74.16\%
Similarity between KO-C4 and KO-C4\_mutated\_19: 75.09\%
Similarity between KO-C4 and KO-C4\_mutated\_20: 74.91\%
Similarity between KO-C5 and KO-C5\_mutated\_1: 75.28\%
Similarity between KO-C5 and KO-C5\_mutated\_2: 76.21\%
Similarity between KO-C5 and KO-C5\_mutated\_3: 76.95\%
Similarity between KO-C5 and KO-C5\_mutated\_4: 75.09\%
Similarity between KO-C5 and KO-C5\_mutated\_5: 74.35\%
Similarity between KO-C5 and KO-C5\_mutated\_6: 75.84\%
Similarity between KO-C5 and KO-C5\_mutated\_7: 75.46\%
Similarity between KO-C5 and KO-C5\_mutated\_8: 75.84\%
Similarity between KO-C5 and KO-C5\_mutated\_9: 75.09\%
Similarity between KO-C5 and KO-C5\_mutated\_10: 74.35\%
Similarity between KO-C5 and KO-C5\_mutated\_11: 75.09\%
Similarity between KO-C5 and KO-C5\_mutated\_12: 74.54\%
Similarity between KO-C5 and KO-C5\_mutated\_13: 74.72\%
Similarity between KO-C5 and KO-C5\_mutated\_14: 74.16\%
Similarity between KO-C5 and KO-C5\_mutated\_15: 74.16\%
Similarity between KO-C5 and KO-C5\_mutated\_16: 75.28\%
Similarity between KO-C5 and KO-C5\_mutated\_17: 76.77\%
Similarity between KO-C5 and KO-C5\_mutated\_18: 75.46\%
Similarity between KO-C5 and KO-C5\_mutated\_19: 75.28\%
Similarity between KO-C5 and KO-C5\_mutated\_20: 74.35\%
Similarity between KO-C6 and KO-C6\_mutated\_1: 75.84\%
Similarity between KO-C6 and KO-C6\_mutated\_2: 76.02\%
Similarity between KO-C6 and KO-C6\_mutated\_3: 74.35\%
Similarity between KO-C6 and KO-C6\_mutated\_4: 74.35\%
Similarity between KO-C6 and KO-C6\_mutated\_5: 74.16\%
Similarity between KO-C6 and KO-C6\_mutated\_6: 74.72\%
Similarity between KO-C6 and KO-C6\_mutated\_7: 75.09\%
Similarity between KO-C6 and KO-C6\_mutated\_8: 75.84\%
Similarity between KO-C6 and KO-C6\_mutated\_9: 74.72\%
Similarity between KO-C6 and KO-C6\_mutated\_10: 74.91\%
Similarity between KO-C6 and KO-C6\_mutated\_11: 75.65\%
Similarity between KO-C6 and KO-C6\_mutated\_12: 75.46\%
Similarity between KO-C6 and KO-C6\_mutated\_13: 74.54\%
Similarity between KO-C6 and KO-C6\_mutated\_14: 75.65\%
Similarity between KO-C6 and KO-C6\_mutated\_15: 76.02\%
Similarity between KO-C6 and KO-C6\_mutated\_16: 75.28\%
Similarity between KO-C6 and KO-C6\_mutated\_17: 76.02\%
Similarity between KO-C6 and KO-C6\_mutated\_18: 74.91\%
Similarity between KO-C6 and KO-C6\_mutated\_19: 75.28\%
Similarity between KO-C6 and KO-C6\_mutated\_20: 74.72\%
Similarity between KO-C7 and KO-C7\_mutated\_1: 75.09\%
Similarity between KO-C7 and KO-C7\_mutated\_2: 75.09\%
Similarity between KO-C7 and KO-C7\_mutated\_3: 74.35\%
Similarity between KO-C7 and KO-C7\_mutated\_4: 74.91\%
Similarity between KO-C7 and KO-C7\_mutated\_5: 72.30\%
Similarity between KO-C7 and KO-C7\_mutated\_6: 74.72\%
Similarity between KO-C7 and KO-C7\_mutated\_7: 75.46\%
Similarity between KO-C7 and KO-C7\_mutated\_8: 76.39\%
Similarity between KO-C7 and KO-C7\_mutated\_9: 75.09\%
Similarity between KO-C7 and KO-C7\_mutated\_10: 75.84\%
Similarity between KO-C7 and KO-C7\_mutated\_11: 75.65\%
Similarity between KO-C7 and KO-C7\_mutated\_12: 74.91\%
Similarity between KO-C7 and KO-C7\_mutated\_13: 75.28\%
Similarity between KO-C7 and KO-C7\_mutated\_14: 74.16\%
Similarity between KO-C7 and KO-C7\_mutated\_15: 74.54\%
Similarity between KO-C7 and KO-C7\_mutated\_16: 73.79\%
Similarity between KO-C7 and KO-C7\_mutated\_17: 74.91\%
Similarity between KO-C7 and KO-C7\_mutated\_18: 74.91\%
Similarity between KO-C7 and KO-C7\_mutated\_19: 75.28\%
Similarity between KO-C7 and KO-C7\_mutated\_20: 75.46\%
Similarity between KO-C8 and KO-C8\_mutated\_1: 75.28\%
Similarity between KO-C8 and KO-C8\_mutated\_2: 75.46\%
Similarity between KO-C8 and KO-C8\_mutated\_3: 75.46\%
Similarity between KO-C8 and KO-C8\_mutated\_4: 74.91\%
Similarity between KO-C8 and KO-C8\_mutated\_5: 74.91\%
Similarity between KO-C8 and KO-C8\_mutated\_6: 74.54\%
Similarity between KO-C8 and KO-C8\_mutated\_7: 74.72\%
Similarity between KO-C8 and KO-C8\_mutated\_8: 74.54\%
Similarity between KO-C8 and KO-C8\_mutated\_9: 73.61\%
Similarity between KO-C8 and KO-C8\_mutated\_10: 75.28\%
Similarity between KO-C8 and KO-C8\_mutated\_11: 74.54\%
Similarity between KO-C8 and KO-C8\_mutated\_12: 74.91\%
Similarity between KO-C8 and KO-C8\_mutated\_13: 73.79\%
Similarity between KO-C8 and KO-C8\_mutated\_14: 74.91\%
Similarity between KO-C8 and KO-C8\_mutated\_15: 74.54\%
Similarity between KO-C8 and KO-C8\_mutated\_16: 75.28\%
Similarity between KO-C8 and KO-C8\_mutated\_17: 74.91\%
Similarity between KO-C8 and KO-C8\_mutated\_18: 74.72\%
Similarity between KO-C8 and KO-C8\_mutated\_19: 74.72\%
Similarity between KO-C8 and KO-C8\_mutated\_20: 73.98\%
Similarity between KO-C9 and KO-C9\_mutated\_1: 74.72\%
Similarity between KO-C9 and KO-C9\_mutated\_2: 75.09\%
Similarity between KO-C9 and KO-C9\_mutated\_3: 76.02\%
Similarity between KO-C9 and KO-C9\_mutated\_4: 75.28\%
Similarity between KO-C9 and KO-C9\_mutated\_5: 74.35\%
Similarity between KO-C9 and KO-C9\_mutated\_6: 73.79\%
Similarity between KO-C9 and KO-C9\_mutated\_7: 74.16\%
Similarity between KO-C9 and KO-C9\_mutated\_8: 76.77\%
Similarity between KO-C9 and KO-C9\_mutated\_9: 74.35\%
Similarity between KO-C9 and KO-C9\_mutated\_10: 74.72\%
Similarity between KO-C9 and KO-C9\_mutated\_11: 74.91\%
Similarity between KO-C9 and KO-C9\_mutated\_12: 75.28\%
Similarity between KO-C9 and KO-C9\_mutated\_13: 75.09\%
Similarity between KO-C9 and KO-C9\_mutated\_14: 75.65\%
Similarity between KO-C9 and KO-C9\_mutated\_15: 74.72\%
Similarity between KO-C9 and KO-C9\_mutated\_16: 73.42\%
Similarity between KO-C9 and KO-C9\_mutated\_17: 74.91\%
Similarity between KO-C9 and KO-C9\_mutated\_18: 75.28\%
Similarity between KO-C9 and KO-C9\_mutated\_19: 76.58\%
Similarity between KO-C9 and KO-C9\_mutated\_20: 75.46\%
Similarity between KO-C10 and KO-C10\_mutated\_1: 75.46\%
Similarity between KO-C10 and KO-C10\_mutated\_2: 76.02\%
Similarity between KO-C10 and KO-C10\_mutated\_3: 73.61\%
Similarity between KO-C10 and KO-C10\_mutated\_4: 75.65\%
Similarity between KO-C10 and KO-C10\_mutated\_5: 75.09\%
Similarity between KO-C10 and KO-C10\_mutated\_6: 75.46\%
Similarity between KO-C10 and KO-C10\_mutated\_7: 76.39\%
Similarity between KO-C10 and KO-C10\_mutated\_8: 75.84\%
Similarity between KO-C10 and KO-C10\_mutated\_9: 74.91\%
Similarity between KO-C10 and KO-C10\_mutated\_10: 76.21\%
Similarity between KO-C10 and KO-C10\_mutated\_11: 74.91\%
Similarity between KO-C10 and KO-C10\_mutated\_12: 75.65\%
Similarity between KO-C10 and KO-C10\_mutated\_13: 75.28\%
Similarity between KO-C10 and KO-C10\_mutated\_14: 74.91\%
Similarity between KO-C10 and KO-C10\_mutated\_15: 75.65\%
Similarity between KO-C10 and KO-C10\_mutated\_16: 75.46\%
Similarity between KO-C10 and KO-C10\_mutated\_17: 76.02\%
Similarity between KO-C10 and KO-C10\_mutated\_18: 75.65\%
Similarity between KO-C10 and KO-C10\_mutated\_19: 75.84\%
Similarity between KO-C10 and KO-C10\_mutated\_20: 73.98\%
Similarity between UZ-TC01 and UZ-TC01\_mutated\_1: 75.09\%
Similarity between UZ-TC01 and UZ-TC01\_mutated\_2: 74.35\%
Similarity between UZ-TC01 and UZ-TC01\_mutated\_3: 75.28\%
Similarity between UZ-TC01 and UZ-TC01\_mutated\_4: 76.58\%
Similarity between UZ-TC01 and UZ-TC01\_mutated\_5: 75.28\%
Similarity between UZ-TC01 and UZ-TC01\_mutated\_6: 77.70\%
Similarity between UZ-TC01 and UZ-TC01\_mutated\_7: 74.91\%
Similarity between UZ-TC01 and UZ-TC01\_mutated\_8: 74.91\%
Similarity between UZ-TC01 and UZ-TC01\_mutated\_9: 74.16\%
Similarity between UZ-TC01 and UZ-TC01\_mutated\_10: 75.65\%
Similarity between UZ-TC01 and UZ-TC01\_mutated\_11: 75.65\%
Similarity between UZ-TC01 and UZ-TC01\_mutated\_12: 73.98\%
Similarity between UZ-TC01 and UZ-TC01\_mutated\_13: 74.72\%
Similarity between UZ-TC01 and UZ-TC01\_mutated\_14: 74.54\%
Similarity between UZ-TC01 and UZ-TC01\_mutated\_15: 74.72\%
Similarity between UZ-TC01 and UZ-TC01\_mutated\_16: 74.72\%
Similarity between UZ-TC01 and UZ-TC01\_mutated\_17: 75.84\%
Similarity between UZ-TC01 and UZ-TC01\_mutated\_18: 74.16\%
Similarity between UZ-TC01 and UZ-TC01\_mutated\_19: 75.46\%
Similarity between UZ-TC01 and UZ-TC01\_mutated\_20: 74.35\%
Similarity between UZ-TC02 and UZ-TC02\_mutated\_1: 74.72\%
Similarity between UZ-TC02 and UZ-TC02\_mutated\_2: 76.02\%
Similarity between UZ-TC02 and UZ-TC02\_mutated\_3: 74.72\%
Similarity between UZ-TC02 and UZ-TC02\_mutated\_4: 74.91\%
Similarity between UZ-TC02 and UZ-TC02\_mutated\_5: 75.46\%
Similarity between UZ-TC02 and UZ-TC02\_mutated\_6: 73.98\%
Similarity between UZ-TC02 and UZ-TC02\_mutated\_7: 74.91\%
Similarity between UZ-TC02 and UZ-TC02\_mutated\_8: 75.84\%
Similarity between UZ-TC02 and UZ-TC02\_mutated\_9: 74.16\%
Similarity between UZ-TC02 and UZ-TC02\_mutated\_10: 73.42\%
Similarity between UZ-TC02 and UZ-TC02\_mutated\_11: 75.65\%
Similarity between UZ-TC02 and UZ-TC02\_mutated\_12: 74.91\%
Similarity between UZ-TC02 and UZ-TC02\_mutated\_13: 75.28\%
Similarity between UZ-TC02 and UZ-TC02\_mutated\_14: 75.65\%
Similarity between UZ-TC02 and UZ-TC02\_mutated\_15: 75.09\%
Similarity between UZ-TC02 and UZ-TC02\_mutated\_16: 75.46\%
Similarity between UZ-TC02 and UZ-TC02\_mutated\_17: 75.28\%
Similarity between UZ-TC02 and UZ-TC02\_mutated\_18: 74.72\%
Similarity between UZ-TC02 and UZ-TC02\_mutated\_19: 73.98\%
Similarity between UZ-TC02 and UZ-TC02\_mutated\_20: 74.91\%
Similarity between UZ-TC03 and UZ-TC03\_mutated\_1: 75.65\%
Similarity between UZ-TC03 and UZ-TC03\_mutated\_2: 75.09\%
Similarity between UZ-TC03 and UZ-TC03\_mutated\_3: 74.35\%
Similarity between UZ-TC03 and UZ-TC03\_mutated\_4: 75.09\%
Similarity between UZ-TC03 and UZ-TC03\_mutated\_5: 74.54\%
Similarity between UZ-TC03 and UZ-TC03\_mutated\_6: 74.91\%
Similarity between UZ-TC03 and UZ-TC03\_mutated\_7: 74.35\%
Similarity between UZ-TC03 and UZ-TC03\_mutated\_8: 73.98\%
Similarity between UZ-TC03 and UZ-TC03\_mutated\_9: 74.54\%
Similarity between UZ-TC03 and UZ-TC03\_mutated\_10: 74.72\%
Similarity between UZ-TC03 and UZ-TC03\_mutated\_11: 75.09\%
Similarity between UZ-TC03 and UZ-TC03\_mutated\_12: 76.02\%
Similarity between UZ-TC03 and UZ-TC03\_mutated\_13: 75.28\%
Similarity between UZ-TC03 and UZ-TC03\_mutated\_14: 76.02\%
Similarity between UZ-TC03 and UZ-TC03\_mutated\_15: 73.79\%
Similarity between UZ-TC03 and UZ-TC03\_mutated\_16: 75.46\%
Similarity between UZ-TC03 and UZ-TC03\_mutated\_17: 74.16\%
Similarity between UZ-TC03 and UZ-TC03\_mutated\_18: 76.39\%
Similarity between UZ-TC03 and UZ-TC03\_mutated\_19: 75.65\%
Similarity between UZ-TC03 and UZ-TC03\_mutated\_20: 75.09\%
    \end{Verbatim}

    \section{Phylogenetic Tree
Construction}\label{phylogenetic-tree-construction}

This Python script uses the BioPython library to construct phylogenetic
trees from a multiple sequence alignment in a FASTA file. The trees are
constructed using the UPGMA (Unweighted Pair Group Method with
Arithmetic Mean) and NJ (Neighbor Joining) methods.

\subsection{Requirements}\label{requirements}

\begin{itemize}
\tightlist
\item
  BioPython
\end{itemize}

\subsection{FILE}\label{file}

    \begin{itemize}
\tightlist
\item
  for output file
\end{itemize}

    \begin{tcolorbox}[breakable, size=fbox, boxrule=1pt, pad at break*=1mm,colback=cellbackground, colframe=cellborder]
\prompt{In}{incolor}{22}{\boxspacing}
\begin{Verbatim}[commandchars=\\\{\}]
\PY{k+kn}{from} \PY{n+nn}{Bio} \PY{k+kn}{import} \PY{n}{Phylo}\PY{p}{,} \PY{n}{AlignIO}
\PY{k+kn}{from} \PY{n+nn}{Bio}\PY{n+nn}{.}\PY{n+nn}{Phylo}\PY{n+nn}{.}\PY{n+nn}{TreeConstruction} \PY{k+kn}{import} \PY{n}{DistanceCalculator}\PY{p}{,} \PY{n}{DistanceTreeConstructor}
\PY{k+kn}{import} \PY{n+nn}{matplotlib}\PY{n+nn}{.}\PY{n+nn}{pyplot} \PY{k}{as} \PY{n+nn}{plt}

\PY{c+c1}{\PYZsh{} Read the sequences from the output fasta file}
\PY{n}{output\PYZus{}fasta\PYZus{}file} \PY{o}{=} \PY{l+s+s1}{\PYZsq{}}\PY{l+s+s1}{output.fasta}\PY{l+s+s1}{\PYZsq{}}
\PY{n}{sequences} \PY{o}{=} \PY{n+nb}{list}\PY{p}{(}\PY{n}{AlignIO}\PY{o}{.}\PY{n}{read}\PY{p}{(}\PY{n}{output\PYZus{}fasta\PYZus{}file}\PY{p}{,} \PY{l+s+s2}{\PYZdq{}}\PY{l+s+s2}{fasta}\PY{l+s+s2}{\PYZdq{}}\PY{p}{)}\PY{p}{)}

\PY{c+c1}{\PYZsh{} Organize sequences into groups: one for each original and its mutations}
\PY{n}{groups} \PY{o}{=} \PY{p}{\PYZob{}}\PY{p}{\PYZcb{}}
\PY{k}{for} \PY{n}{seq} \PY{o+ow}{in} \PY{n}{sequences}\PY{p}{:}
    \PY{n}{group\PYZus{}id} \PY{o}{=} \PY{n}{seq}\PY{o}{.}\PY{n}{id}\PY{o}{.}\PY{n}{split}\PY{p}{(}\PY{l+s+s1}{\PYZsq{}}\PY{l+s+s1}{\PYZus{}mutated}\PY{l+s+s1}{\PYZsq{}}\PY{p}{)}\PY{p}{[}\PY{l+m+mi}{0}\PY{p}{]}  \PY{c+c1}{\PYZsh{} Get the base ID for grouping}
    \PY{k}{if} \PY{n}{group\PYZus{}id} \PY{o+ow}{not} \PY{o+ow}{in} \PY{n}{groups}\PY{p}{:}
        \PY{n}{groups}\PY{p}{[}\PY{n}{group\PYZus{}id}\PY{p}{]} \PY{o}{=} \PY{p}{[}\PY{p}{]}
    \PY{n}{groups}\PY{p}{[}\PY{n}{group\PYZus{}id}\PY{p}{]}\PY{o}{.}\PY{n}{append}\PY{p}{(}\PY{n}{seq}\PY{p}{)}

\PY{c+c1}{\PYZsh{} Calculate the distance matrix, construct trees, and draw them for each group}
\PY{n}{calculator} \PY{o}{=} \PY{n}{DistanceCalculator}\PY{p}{(}\PY{l+s+s1}{\PYZsq{}}\PY{l+s+s1}{identity}\PY{l+s+s1}{\PYZsq{}}\PY{p}{)}
\PY{n}{constructor} \PY{o}{=} \PY{n}{DistanceTreeConstructor}\PY{p}{(}\PY{p}{)}

\PY{c+c1}{\PYZsh{} Set up matplotlib figures to display trees}
\PY{c+c1}{\PYZsh{} Each group needs two plots, one for UPGMA and one for NJ}
\PY{n}{fig}\PY{p}{,} \PY{n}{axes} \PY{o}{=} \PY{n}{plt}\PY{o}{.}\PY{n}{subplots}\PY{p}{(}\PY{n}{nrows}\PY{o}{=}\PY{n+nb}{len}\PY{p}{(}\PY{n}{groups}\PY{p}{)}\PY{o}{*}\PY{l+m+mi}{2}\PY{p}{,} \PY{n}{figsize}\PY{o}{=}\PY{p}{(}\PY{l+m+mi}{10}\PY{p}{,} \PY{l+m+mi}{5} \PY{o}{*} \PY{n+nb}{len}\PY{p}{(}\PY{n}{groups}\PY{p}{)}\PY{o}{*}\PY{l+m+mi}{2}\PY{p}{)}\PY{p}{)}  \PY{c+c1}{\PYZsh{} Adjust size as needed}

\PY{c+c1}{\PYZsh{} Handle case with only one group separately}
\PY{n}{single\PYZus{}group} \PY{o}{=} \PY{n+nb}{len}\PY{p}{(}\PY{n}{groups}\PY{p}{)} \PY{o}{==} \PY{l+m+mi}{1}

\PY{k}{for} \PY{n}{idx}\PY{p}{,} \PY{p}{(}\PY{n}{group\PYZus{}id}\PY{p}{,} \PY{n}{group\PYZus{}seqs}\PY{p}{)} \PY{o+ow}{in} \PY{n+nb}{enumerate}\PY{p}{(}\PY{n}{groups}\PY{o}{.}\PY{n}{items}\PY{p}{(}\PY{p}{)}\PY{p}{)}\PY{p}{:}
    \PY{n}{alignment} \PY{o}{=} \PY{n}{AlignIO}\PY{o}{.}\PY{n}{MultipleSeqAlignment}\PY{p}{(}\PY{n}{group\PYZus{}seqs}\PY{p}{)}
    \PY{n}{dm} \PY{o}{=} \PY{n}{calculator}\PY{o}{.}\PY{n}{get\PYZus{}distance}\PY{p}{(}\PY{n}{alignment}\PY{p}{)}
    \PY{n}{upgma\PYZus{}tree} \PY{o}{=} \PY{n}{constructor}\PY{o}{.}\PY{n}{upgma}\PY{p}{(}\PY{n}{dm}\PY{p}{)}
    \PY{n}{nj\PYZus{}tree} \PY{o}{=} \PY{n}{constructor}\PY{o}{.}\PY{n}{nj}\PY{p}{(}\PY{n}{dm}\PY{p}{)}

    \PY{c+c1}{\PYZsh{} Get the correct axes for the UPGMA and NJ trees}
    \PY{n}{ax1} \PY{o}{=} \PY{n}{axes}\PY{p}{[}\PY{n}{idx}\PY{o}{*}\PY{l+m+mi}{2}\PY{p}{]} \PY{k}{if} \PY{o+ow}{not} \PY{n}{single\PYZus{}group} \PY{k}{else} \PY{n}{axes}
    \PY{n}{ax2} \PY{o}{=} \PY{n}{axes}\PY{p}{[}\PY{n}{idx}\PY{o}{*}\PY{l+m+mi}{2} \PY{o}{+} \PY{l+m+mi}{1}\PY{p}{]} \PY{k}{if} \PY{o+ow}{not} \PY{n}{single\PYZus{}group} \PY{k}{else} \PY{n}{axes}

    \PY{c+c1}{\PYZsh{} Draw UPGMA tree}
    \PY{n}{ax1}\PY{o}{.}\PY{n}{set\PYZus{}title}\PY{p}{(}\PY{l+s+sa}{f}\PY{l+s+s1}{\PYZsq{}}\PY{l+s+s1}{UPGMA Tree for }\PY{l+s+si}{\PYZob{}}\PY{n}{group\PYZus{}id}\PY{l+s+si}{\PYZcb{}}\PY{l+s+s1}{\PYZsq{}}\PY{p}{)}
    \PY{n}{Phylo}\PY{o}{.}\PY{n}{draw}\PY{p}{(}\PY{n}{upgma\PYZus{}tree}\PY{p}{,} \PY{n}{do\PYZus{}show}\PY{o}{=}\PY{k+kc}{False}\PY{p}{,} \PY{n}{axes}\PY{o}{=}\PY{n}{ax1}\PY{p}{)}

    \PY{c+c1}{\PYZsh{} Draw NJ tree}
    \PY{n}{ax2}\PY{o}{.}\PY{n}{set\PYZus{}title}\PY{p}{(}\PY{l+s+sa}{f}\PY{l+s+s1}{\PYZsq{}}\PY{l+s+s1}{NJ Tree for }\PY{l+s+si}{\PYZob{}}\PY{n}{group\PYZus{}id}\PY{l+s+si}{\PYZcb{}}\PY{l+s+s1}{\PYZsq{}}\PY{p}{)}
    \PY{n}{Phylo}\PY{o}{.}\PY{n}{draw}\PY{p}{(}\PY{n}{nj\PYZus{}tree}\PY{p}{,} \PY{n}{do\PYZus{}show}\PY{o}{=}\PY{k+kc}{False}\PY{p}{,} \PY{n}{axes}\PY{o}{=}\PY{n}{ax2}\PY{p}{)}

\PY{n}{plt}\PY{o}{.}\PY{n}{tight\PYZus{}layout}\PY{p}{(}\PY{p}{)}
\PY{n}{plt}\PY{o}{.}\PY{n}{show}\PY{p}{(}\PY{p}{)}
\end{Verbatim}
\end{tcolorbox}

    \begin{center}
    \adjustimage{max size={0.9\linewidth}{0.9\paperheight}}{ass2_files/ass2_11_0.png}
    \end{center}
    { \hspace*{\fill} \\}
    
    \begin{itemize}
\tightlist
\item
  input
\end{itemize}

    \begin{tcolorbox}[breakable, size=fbox, boxrule=1pt, pad at break*=1mm,colback=cellbackground, colframe=cellborder]
\prompt{In}{incolor}{23}{\boxspacing}
\begin{Verbatim}[commandchars=\\\{\}]
\PY{k+kn}{from} \PY{n+nn}{Bio} \PY{k+kn}{import} \PY{n}{Phylo}\PY{p}{,} \PY{n}{AlignIO}
\PY{k+kn}{from} \PY{n+nn}{Bio}\PY{n+nn}{.}\PY{n+nn}{Phylo}\PY{n+nn}{.}\PY{n+nn}{TreeConstruction} \PY{k+kn}{import} \PY{n}{DistanceCalculator}\PY{p}{,} \PY{n}{DistanceTreeConstructor}

\PY{c+c1}{\PYZsh{} Read the fasta file and convert it into a dictionary}
\PY{n}{alignment} \PY{o}{=} \PY{n}{AlignIO}\PY{o}{.}\PY{n}{read}\PY{p}{(}\PY{l+s+s1}{\PYZsq{}}\PY{l+s+s1}{input.fasta}\PY{l+s+s1}{\PYZsq{}}\PY{p}{,} \PY{l+s+s1}{\PYZsq{}}\PY{l+s+s1}{fasta}\PY{l+s+s1}{\PYZsq{}}\PY{p}{)}

\PY{c+c1}{\PYZsh{} Calculate the distance matrix}
\PY{n}{calculator} \PY{o}{=} \PY{n}{DistanceCalculator}\PY{p}{(}\PY{l+s+s1}{\PYZsq{}}\PY{l+s+s1}{identity}\PY{l+s+s1}{\PYZsq{}}\PY{p}{)}
\PY{n}{dm} \PY{o}{=} \PY{n}{calculator}\PY{o}{.}\PY{n}{get\PYZus{}distance}\PY{p}{(}\PY{n}{alignment}\PY{p}{)}

\PY{c+c1}{\PYZsh{} Construct the phylogenetic tree using the UPGMA method}
\PY{n}{constructor} \PY{o}{=} \PY{n}{DistanceTreeConstructor}\PY{p}{(}\PY{p}{)}
\PY{n}{upgma\PYZus{}tree} \PY{o}{=} \PY{n}{constructor}\PY{o}{.}\PY{n}{upgma}\PY{p}{(}\PY{n}{dm}\PY{p}{)}

\PY{c+c1}{\PYZsh{} Construct the phylogenetic tree using the Neighbor Joining method}
\PY{n}{nj\PYZus{}tree} \PY{o}{=} \PY{n}{constructor}\PY{o}{.}\PY{n}{nj}\PY{p}{(}\PY{n}{dm}\PY{p}{)}

\PY{c+c1}{\PYZsh{} Draw the UPGMA and NJ trees}
\PY{n}{Phylo}\PY{o}{.}\PY{n}{draw}\PY{p}{(}\PY{n}{upgma\PYZus{}tree}\PY{p}{)}
\PY{n}{Phylo}\PY{o}{.}\PY{n}{draw}\PY{p}{(}\PY{n}{nj\PYZus{}tree}\PY{p}{)}
\end{Verbatim}
\end{tcolorbox}

    \begin{center}
    \adjustimage{max size={0.9\linewidth}{0.9\paperheight}}{ass2_files/ass2_13_0.png}
    \end{center}
    { \hspace*{\fill} \\}
    
    \begin{center}
    \adjustimage{max size={0.9\linewidth}{0.9\paperheight}}{ass2_files/ass2_13_1.png}
    \end{center}
    { \hspace*{\fill} \\}
    
    \subsection{Explanation}\label{explanation}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  The script first reads a multiple sequence alignment from a FASTA file
  using the \texttt{AlignIO.read} function.
\item
  It then calculates a distance matrix from the alignment using the
  \texttt{DistanceCalculator} class with the `identity' model, which
  calculates the proportion of mismatches between sequences.
\item
  The script constructs a phylogenetic tree from the distance matrix
  using the UPGMA method, which clusters sequences based on their
  pairwise distances.
\item
  The script also constructs a phylogenetic tree using the NJ method,
  which also uses pairwise distances but takes a different approach to
  clustering.
\item
  Finally, the script draws the UPGMA and NJ trees using the
  \texttt{Phylo.draw} function.
\end{enumerate}

    \section{Phylogenetic Tree
Visualization}\label{phylogenetic-tree-visualization}

This Python script uses the BioPython and ETE Toolkit libraries to
visualize phylogenetic trees. The trees are first constructed using the
UPGMA (Unweighted Pair Group Method with Arithmetic Mean) and NJ
(Neighbor Joining) methods, and then visualized using the ETE Toolkit.

\subsection{Requirements}\label{requirements}

\begin{itemize}
\tightlist
\item
  BioPython
\item
  ETE Toolkit
\end{itemize}

    \begin{tcolorbox}[breakable, size=fbox, boxrule=1pt, pad at break*=1mm,colback=cellbackground, colframe=cellborder]
\prompt{In}{incolor}{24}{\boxspacing}
\begin{Verbatim}[commandchars=\\\{\}]
\PY{k+kn}{from} \PY{n+nn}{ete3} \PY{k+kn}{import} \PY{n}{Tree}\PY{p}{,} \PY{n}{TreeStyle}
\PY{k+kn}{from} \PY{n+nn}{io} \PY{k+kn}{import} \PY{n}{StringIO}
\PY{k+kn}{from} \PY{n+nn}{Bio} \PY{k+kn}{import} \PY{n}{Phylo}

\PY{c+c1}{\PYZsh{} Add a root name to the trees}
\PY{n}{upgma\PYZus{}tree}\PY{o}{.}\PY{n}{root}\PY{o}{.}\PY{n}{name} \PY{o}{=} \PY{l+s+s2}{\PYZdq{}}\PY{l+s+s2}{Root}\PY{l+s+s2}{\PYZdq{}}
\PY{n}{nj\PYZus{}tree}\PY{o}{.}\PY{n}{root}\PY{o}{.}\PY{n}{name} \PY{o}{=} \PY{l+s+s2}{\PYZdq{}}\PY{l+s+s2}{Root}\PY{l+s+s2}{\PYZdq{}}

\PY{c+c1}{\PYZsh{} Remove internal node names}
\PY{k}{for} \PY{n}{clade} \PY{o+ow}{in} \PY{n}{upgma\PYZus{}tree}\PY{o}{.}\PY{n}{find\PYZus{}clades}\PY{p}{(}\PY{p}{)}\PY{p}{:}
    \PY{k}{if} \PY{o+ow}{not} \PY{n}{clade}\PY{o}{.}\PY{n}{is\PYZus{}terminal}\PY{p}{(}\PY{p}{)}\PY{p}{:}
        \PY{n}{clade}\PY{o}{.}\PY{n}{name} \PY{o}{=} \PY{l+s+s2}{\PYZdq{}}\PY{l+s+s2}{\PYZdq{}}
\PY{k}{for} \PY{n}{clade} \PY{o+ow}{in} \PY{n}{nj\PYZus{}tree}\PY{o}{.}\PY{n}{find\PYZus{}clades}\PY{p}{(}\PY{p}{)}\PY{p}{:}
    \PY{k}{if} \PY{o+ow}{not} \PY{n}{clade}\PY{o}{.}\PY{n}{is\PYZus{}terminal}\PY{p}{(}\PY{p}{)}\PY{p}{:}
        \PY{n}{clade}\PY{o}{.}\PY{n}{name} \PY{o}{=} \PY{l+s+s2}{\PYZdq{}}\PY{l+s+s2}{\PYZdq{}}

\PY{c+c1}{\PYZsh{} Convert the Bio.Phylo trees to Newick format and then to ETE trees}
\PY{n}{upgma\PYZus{}newick} \PY{o}{=} \PY{n}{StringIO}\PY{p}{(}\PY{p}{)}
\PY{n}{Phylo}\PY{o}{.}\PY{n}{write}\PY{p}{(}\PY{n}{upgma\PYZus{}tree}\PY{p}{,} \PY{n}{upgma\PYZus{}newick}\PY{p}{,} \PY{n+nb}{format}\PY{o}{=}\PY{l+s+s2}{\PYZdq{}}\PY{l+s+s2}{newick}\PY{l+s+s2}{\PYZdq{}}\PY{p}{)}
\PY{n}{upgma\PYZus{}ete\PYZus{}tree} \PY{o}{=} \PY{n}{Tree}\PY{p}{(}\PY{n}{upgma\PYZus{}newick}\PY{o}{.}\PY{n}{getvalue}\PY{p}{(}\PY{p}{)}\PY{p}{)}

\PY{n}{nj\PYZus{}newick} \PY{o}{=} \PY{n}{StringIO}\PY{p}{(}\PY{p}{)}
\PY{n}{Phylo}\PY{o}{.}\PY{n}{write}\PY{p}{(}\PY{n}{nj\PYZus{}tree}\PY{p}{,} \PY{n}{nj\PYZus{}newick}\PY{p}{,} \PY{n+nb}{format}\PY{o}{=}\PY{l+s+s2}{\PYZdq{}}\PY{l+s+s2}{newick}\PY{l+s+s2}{\PYZdq{}}\PY{p}{)}
\PY{n}{nj\PYZus{}ete\PYZus{}tree} \PY{o}{=} \PY{n}{Tree}\PY{p}{(}\PY{n}{nj\PYZus{}newick}\PY{o}{.}\PY{n}{getvalue}\PY{p}{(}\PY{p}{)}\PY{p}{)}

\PY{c+c1}{\PYZsh{} Create a TreeStyle object for customizing the tree\PYZsq{}s appearance}
\PY{n}{ts} \PY{o}{=} \PY{n}{TreeStyle}\PY{p}{(}\PY{p}{)}
\PY{n}{ts}\PY{o}{.}\PY{n}{show\PYZus{}leaf\PYZus{}name} \PY{o}{=} \PY{k+kc}{True}
\PY{n}{ts}\PY{o}{.}\PY{n}{show\PYZus{}branch\PYZus{}length} \PY{o}{=} \PY{k+kc}{True}
\PY{n}{ts}\PY{o}{.}\PY{n}{show\PYZus{}branch\PYZus{}support} \PY{o}{=} \PY{k+kc}{True}

\PY{c+c1}{\PYZsh{} Show the trees}
\PY{n}{upgma\PYZus{}ete\PYZus{}tree}\PY{o}{.}\PY{n}{show}\PY{p}{(}\PY{n}{tree\PYZus{}style}\PY{o}{=}\PY{n}{ts}\PY{p}{)}
\PY{n}{nj\PYZus{}ete\PYZus{}tree}\PY{o}{.}\PY{n}{show}\PY{p}{(}\PY{n}{tree\PYZus{}style}\PY{o}{=}\PY{n}{ts}\PY{p}{)}
\end{Verbatim}
\end{tcolorbox}

    \subsection{Explanation}\label{explanation}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  The script first adds a root name to the UPGMA and NJ trees.
\item
  It then removes the internal node names from the trees. This is done
  because the ETE Toolkit does not support internal node names in the
  Newick format.
\item
  The script converts the Bio.Phylo trees to Newick format and then to
  ETE trees. This is done using the \texttt{Phylo.write} function and
  the \texttt{Tree} class from the ETE Toolkit.
\item
  The script creates a \texttt{TreeStyle} object to customize the
  appearance of the trees. It sets the \texttt{show\_leaf\_name},
  \texttt{show\_branch\_length}, and \texttt{show\_branch\_support}
  attributes to \texttt{True} to display the leaf names, branch lengths,
  and branch support values, respectively.
\item
  Finally, the script displays the trees using the \texttt{show} method
  of the \texttt{Tree} class. The \texttt{tree\_style} parameter is set
  to the \texttt{TreeStyle} object to apply the custom appearance.
\end{enumerate}


    % Add a bibliography block to the postdoc
    
    
    
\end{document}
